// @storefront-ui/tailwind-config/dist/index.mjs
import Un from "@mertasan/tailwindcss-variables";
import $n from "@storefront-ui/tw-plugin-peer-next";
var Ue = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {};
function Hn(a) {
  return a && a.__esModule && Object.prototype.hasOwnProperty.call(a, "default") ? a.default : a;
}
var cr = {}, pr = {};
(function(a) {
  Object.defineProperty(a, "__esModule", {
    value: !0
  }), Object.defineProperty(a, "default", {
    enumerable: !0,
    get: function() {
      return b;
    }
  });
  function t(d, g) {
    return {
      handler: d,
      config: g
    };
  }
  t.withOptions = function(d, g = () => ({})) {
    const y = function(p) {
      return {
        __options: p,
        handler: d(p),
        config: g(p)
      };
    };
    return y.__isOptionsFunction = !0, y.__pluginFunction = d, y.__configFunction = g, y;
  };
  const b = t;
})(pr);
(function(a) {
  Object.defineProperty(a, "__esModule", {
    value: !0
  }), Object.defineProperty(a, "default", {
    enumerable: !0,
    get: function() {
      return d;
    }
  });
  const t = /* @__PURE__ */ b(pr);
  function b(g) {
    return g && g.__esModule ? g : {
      default: g
    };
  }
  const d = t.default;
})(cr);
let ht = cr;
var Gn = (ht.__esModule ? ht : { default: ht }).default, et = { exports: {} };
et.exports;
(function(a, t) {
  var b = 200, d = "__lodash_hash_undefined__", g = 800, y = 16, p = 9007199254740991, m = "[object Arguments]", n = "[object Array]", o = "[object AsyncFunction]", l = "[object Boolean]", f = "[object Date]", w = "[object Error]", A = "[object Function]", R = "[object GeneratorFunction]", i = "[object Map]", h = "[object Number]", S = "[object Null]", E = "[object Object]", O = "[object Proxy]", x = "[object RegExp]", T = "[object Set]", P = "[object String]", C = "[object Undefined]", B = "[object WeakMap]", I = "[object ArrayBuffer]", D = "[object DataView]", Y = "[object Float32Array]", Z = "[object Float64Array]", pe = "[object Int8Array]", ue = "[object Int16Array]", fe = "[object Int32Array]", j = "[object Uint8Array]", he = "[object Uint8ClampedArray]", Se = "[object Uint16Array]", Re = "[object Uint32Array]", Oe = /[\\^$.*+?()[\]{}|]/g, F = /^\[object .+?Constructor\]$/, N = /^(?:0|[1-9]\d*)$/, k = {};
  k[Y] = k[Z] = k[pe] = k[ue] = k[fe] = k[j] = k[he] = k[Se] = k[Re] = !0, k[m] = k[n] = k[I] = k[l] = k[D] = k[f] = k[w] = k[A] = k[i] = k[h] = k[E] = k[x] = k[T] = k[P] = k[B] = !1;
  var v = typeof Ue == "object" && Ue && Ue.Object === Object && Ue, L = typeof self == "object" && self && self.Object === Object && self, z = v || L || Function("return this")(), _ = t && !t.nodeType && t, $ = _ && !0 && a && !a.nodeType && a, J = $ && $.exports === _, Q = J && v.process, W = function() {
    try {
      var r = $ && $.require && $.require("util").types;
      return r || Q && Q.binding && Q.binding("util");
    } catch {
    }
  }(), ee = W && W.isTypedArray;
  function le(r, u, c) {
    switch (c.length) {
      case 0:
        return r.call(u);
      case 1:
        return r.call(u, c[0]);
      case 2:
        return r.call(u, c[0], c[1]);
      case 3:
        return r.call(u, c[0], c[1], c[2]);
    }
    return r.apply(u, c);
  }
  function H(r, u) {
    for (var c = -1, q = Array(r); ++c < r; )
      q[c] = u(c);
    return q;
  }
  function ie(r) {
    return function(u) {
      return r(u);
    };
  }
  function Le(r, u) {
    return r == null ? void 0 : r[u];
  }
  function Ee(r, u) {
    return function(c) {
      return r(u(c));
    };
  }
  var ge = Array.prototype, ce = Function.prototype, de = Object.prototype, ve = z["__core-js_shared__"], _e = ce.toString, we = de.hasOwnProperty, Fe = function() {
    var r = /[^.]+$/.exec(ve && ve.keys && ve.keys.IE_PROTO || "");
    return r ? "Symbol(src)_1." + r : "";
  }(), Be = de.toString, rt = _e.call(Object), nt = RegExp(
    "^" + _e.call(we).replace(Oe, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
  ), Ce = J ? z.Buffer : void 0, je = z.Symbol, Qe = z.Uint8Array, Ye = Ce ? Ce.allocUnsafe : void 0, Vt = Ee(Object.getPrototypeOf, Object), Kt = Object.create, Mr = de.propertyIsEnumerable, Nr = ge.splice, qe = je ? je.toStringTag : void 0, Ve = function() {
    try {
      var r = at(Object, "defineProperty");
      return r({}, "", {}), r;
    } catch {
    }
  }(), Cr = Ce ? Ce.isBuffer : void 0, Jt = Math.max, zr = Date.now, Xt = at(z, "Map"), $e = at(Object, "create"), Wr = function() {
    function r() {
    }
    return function(u) {
      if (!Ne(u))
        return {};
      if (Kt)
        return Kt(u);
      r.prototype = u;
      var c = new r();
      return r.prototype = void 0, c;
    };
  }();
  function Me(r) {
    var u = -1, c = r == null ? 0 : r.length;
    for (this.clear(); ++u < c; ) {
      var q = r[u];
      this.set(q[0], q[1]);
    }
  }
  function Ur() {
    this.__data__ = $e ? $e(null) : {}, this.size = 0;
  }
  function $r(r) {
    var u = this.has(r) && delete this.__data__[r];
    return this.size -= u ? 1 : 0, u;
  }
  function Hr(r) {
    var u = this.__data__;
    if ($e) {
      var c = u[r];
      return c === d ? void 0 : c;
    }
    return we.call(u, r) ? u[r] : void 0;
  }
  function Gr(r) {
    var u = this.__data__;
    return $e ? u[r] !== void 0 : we.call(u, r);
  }
  function jr(r, u) {
    var c = this.__data__;
    return this.size += this.has(r) ? 0 : 1, c[r] = $e && u === void 0 ? d : u, this;
  }
  Me.prototype.clear = Ur, Me.prototype.delete = $r, Me.prototype.get = Hr, Me.prototype.has = Gr, Me.prototype.set = jr;
  function Ae(r) {
    var u = -1, c = r == null ? 0 : r.length;
    for (this.clear(); ++u < c; ) {
      var q = r[u];
      this.set(q[0], q[1]);
    }
  }
  function Qr() {
    this.__data__ = [], this.size = 0;
  }
  function Yr(r) {
    var u = this.__data__, c = Ke(u, r);
    if (c < 0)
      return !1;
    var q = u.length - 1;
    return c == q ? u.pop() : Nr.call(u, c, 1), --this.size, !0;
  }
  function Vr(r) {
    var u = this.__data__, c = Ke(u, r);
    return c < 0 ? void 0 : u[c][1];
  }
  function Kr(r) {
    return Ke(this.__data__, r) > -1;
  }
  function Jr(r, u) {
    var c = this.__data__, q = Ke(c, r);
    return q < 0 ? (++this.size, c.push([r, u])) : c[q][1] = u, this;
  }
  Ae.prototype.clear = Qr, Ae.prototype.delete = Yr, Ae.prototype.get = Vr, Ae.prototype.has = Kr, Ae.prototype.set = Jr;
  function ze(r) {
    var u = -1, c = r == null ? 0 : r.length;
    for (this.clear(); ++u < c; ) {
      var q = r[u];
      this.set(q[0], q[1]);
    }
  }
  function Xr() {
    this.size = 0, this.__data__ = {
      hash: new Me(),
      map: new (Xt || Ae)(),
      string: new Me()
    };
  }
  function Zr(r) {
    var u = Xe(this, r).delete(r);
    return this.size -= u ? 1 : 0, u;
  }
  function en(r) {
    return Xe(this, r).get(r);
  }
  function tn(r) {
    return Xe(this, r).has(r);
  }
  function rn(r, u) {
    var c = Xe(this, r), q = c.size;
    return c.set(r, u), this.size += c.size == q ? 0 : 1, this;
  }
  ze.prototype.clear = Xr, ze.prototype.delete = Zr, ze.prototype.get = en, ze.prototype.has = tn, ze.prototype.set = rn;
  function We(r) {
    var u = this.__data__ = new Ae(r);
    this.size = u.size;
  }
  function nn() {
    this.__data__ = new Ae(), this.size = 0;
  }
  function on(r) {
    var u = this.__data__, c = u.delete(r);
    return this.size = u.size, c;
  }
  function an(r) {
    return this.__data__.get(r);
  }
  function sn(r) {
    return this.__data__.has(r);
  }
  function un(r, u) {
    var c = this.__data__;
    if (c instanceof Ae) {
      var q = c.__data__;
      if (!Xt || q.length < b - 1)
        return q.push([r, u]), this.size = ++c.size, this;
      c = this.__data__ = new ze(q);
    }
    return c.set(r, u), this.size = c.size, this;
  }
  We.prototype.clear = nn, We.prototype.delete = on, We.prototype.get = an, We.prototype.has = sn, We.prototype.set = un;
  function ln(r, u) {
    var c = lt(r), q = !c && ut(r), K = !c && !q && nr(r), te = !c && !q && !K && or(r), oe = c || q || K || te, V = oe ? H(r.length, String) : [], ae = V.length;
    for (var ke in r)
      (u || we.call(r, ke)) && !(oe && // Safari 9 has enumerable `arguments.length` in strict mode.
      (ke == "length" || // Node.js 0.10 has enumerable non-index properties on buffers.
      K && (ke == "offset" || ke == "parent") || // PhantomJS 2 has enumerable non-index properties on typed arrays.
      te && (ke == "buffer" || ke == "byteLength" || ke == "byteOffset") || // Skip index properties.
      tr(ke, ae))) && V.push(ke);
    return V;
  }
  function it(r, u, c) {
    (c !== void 0 && !Ze(r[u], c) || c === void 0 && !(u in r)) && ot(r, u, c);
  }
  function fn(r, u, c) {
    var q = r[u];
    (!(we.call(r, u) && Ze(q, c)) || c === void 0 && !(u in r)) && ot(r, u, c);
  }
  function Ke(r, u) {
    for (var c = r.length; c--; )
      if (Ze(r[c][0], u))
        return c;
    return -1;
  }
  function ot(r, u, c) {
    u == "__proto__" && Ve ? Ve(r, u, {
      configurable: !0,
      enumerable: !0,
      value: c,
      writable: !0
    }) : r[u] = c;
  }
  var cn = On();
  function Je(r) {
    return r == null ? r === void 0 ? C : S : qe && qe in Object(r) ? En(r) : An(r);
  }
  function Zt(r) {
    return He(r) && Je(r) == m;
  }
  function pn(r) {
    if (!Ne(r) || Ln(r))
      return !1;
    var u = ct(r) ? nt : F;
    return u.test(Fn(r));
  }
  function dn(r) {
    return He(r) && ir(r.length) && !!k[Je(r)];
  }
  function hn(r) {
    if (!Ne(r))
      return In(r);
    var u = rr(r), c = [];
    for (var q in r)
      q == "constructor" && (u || !we.call(r, q)) || c.push(q);
    return c;
  }
  function er(r, u, c, q, K) {
    r !== u && cn(u, function(te, oe) {
      if (K || (K = new We()), Ne(te))
        gn(r, u, oe, c, er, q, K);
      else {
        var V = q ? q(st(r, oe), te, oe + "", r, u, K) : void 0;
        V === void 0 && (V = te), it(r, oe, V);
      }
    }, ar);
  }
  function gn(r, u, c, q, K, te, oe) {
    var V = st(r, c), ae = st(u, c), ke = oe.get(ae);
    if (ke) {
      it(r, c, ke);
      return;
    }
    var Te = te ? te(V, ae, c + "", r, u, oe) : void 0, Ge = Te === void 0;
    if (Ge) {
      var pt = lt(ae), dt = !pt && nr(ae), ur = !pt && !dt && or(ae);
      Te = ae, pt || dt || ur ? lt(V) ? Te = V : qn(V) ? Te = Tn(V) : dt ? (Ge = !1, Te = wn(ae, !0)) : ur ? (Ge = !1, Te = _n(ae, !0)) : Te = [] : Mn(ae) || ut(ae) ? (Te = V, ut(V) ? Te = Nn(V) : (!Ne(V) || ct(V)) && (Te = kn(ae))) : Ge = !1;
    }
    Ge && (oe.set(ae, Te), K(Te, ae, q, te, oe), oe.delete(ae)), it(r, c, Te);
  }
  function vn(r, u) {
    return Dn(Bn(r, u, sr), r + "");
  }
  var mn = Ve ? function(r, u) {
    return Ve(r, "toString", {
      configurable: !0,
      enumerable: !1,
      value: zn(u),
      writable: !0
    });
  } : sr;
  function wn(r, u) {
    if (u)
      return r.slice();
    var c = r.length, q = Ye ? Ye(c) : new r.constructor(c);
    return r.copy(q), q;
  }
  function bn(r) {
    var u = new r.constructor(r.byteLength);
    return new Qe(u).set(new Qe(r)), u;
  }
  function _n(r, u) {
    var c = u ? bn(r.buffer) : r.buffer;
    return new r.constructor(c, r.byteOffset, r.length);
  }
  function Tn(r, u) {
    var c = -1, q = r.length;
    for (u || (u = Array(q)); ++c < q; )
      u[c] = r[c];
    return u;
  }
  function yn(r, u, c, q) {
    var K = !c;
    c || (c = {});
    for (var te = -1, oe = u.length; ++te < oe; ) {
      var V = u[te], ae = q ? q(c[V], r[V], V, c, r) : void 0;
      ae === void 0 && (ae = r[V]), K ? ot(c, V, ae) : fn(c, V, ae);
    }
    return c;
  }
  function Sn(r) {
    return vn(function(u, c) {
      var q = -1, K = c.length, te = K > 1 ? c[K - 1] : void 0, oe = K > 2 ? c[2] : void 0;
      for (te = r.length > 3 && typeof te == "function" ? (K--, te) : void 0, oe && xn(c[0], c[1], oe) && (te = K < 3 ? void 0 : te, K = 1), u = Object(u); ++q < K; ) {
        var V = c[q];
        V && r(u, V, q, te);
      }
      return u;
    });
  }
  function On(r) {
    return function(u, c, q) {
      for (var K = -1, te = Object(u), oe = q(u), V = oe.length; V--; ) {
        var ae = oe[r ? V : ++K];
        if (c(te[ae], ae, te) === !1)
          break;
      }
      return u;
    };
  }
  function Xe(r, u) {
    var c = r.__data__;
    return Pn(u) ? c[typeof u == "string" ? "string" : "hash"] : c.map;
  }
  function at(r, u) {
    var c = Le(r, u);
    return pn(c) ? c : void 0;
  }
  function En(r) {
    var u = we.call(r, qe), c = r[qe];
    try {
      r[qe] = void 0;
      var q = !0;
    } catch {
    }
    var K = Be.call(r);
    return q && (u ? r[qe] = c : delete r[qe]), K;
  }
  function kn(r) {
    return typeof r.constructor == "function" && !rr(r) ? Wr(Vt(r)) : {};
  }
  function tr(r, u) {
    var c = typeof r;
    return u = u ?? p, !!u && (c == "number" || c != "symbol" && N.test(r)) && r > -1 && r % 1 == 0 && r < u;
  }
  function xn(r, u, c) {
    if (!Ne(c))
      return !1;
    var q = typeof u;
    return (q == "number" ? ft(c) && tr(u, c.length) : q == "string" && u in c) ? Ze(c[u], r) : !1;
  }
  function Pn(r) {
    var u = typeof r;
    return u == "string" || u == "number" || u == "symbol" || u == "boolean" ? r !== "__proto__" : r === null;
  }
  function Ln(r) {
    return !!Fe && Fe in r;
  }
  function rr(r) {
    var u = r && r.constructor, c = typeof u == "function" && u.prototype || de;
    return r === c;
  }
  function In(r) {
    var u = [];
    if (r != null)
      for (var c in Object(r))
        u.push(c);
    return u;
  }
  function An(r) {
    return Be.call(r);
  }
  function Bn(r, u, c) {
    return u = Jt(u === void 0 ? r.length - 1 : u, 0), function() {
      for (var q = arguments, K = -1, te = Jt(q.length - u, 0), oe = Array(te); ++K < te; )
        oe[K] = q[u + K];
      K = -1;
      for (var V = Array(u + 1); ++K < u; )
        V[K] = q[K];
      return V[u] = c(oe), le(r, this, V);
    };
  }
  function st(r, u) {
    if (!(u === "constructor" && typeof r[u] == "function") && u != "__proto__")
      return r[u];
  }
  var Dn = Rn(mn);
  function Rn(r) {
    var u = 0, c = 0;
    return function() {
      var q = zr(), K = y - (q - c);
      if (c = q, K > 0) {
        if (++u >= g)
          return arguments[0];
      } else
        u = 0;
      return r.apply(void 0, arguments);
    };
  }
  function Fn(r) {
    if (r != null) {
      try {
        return _e.call(r);
      } catch {
      }
      try {
        return r + "";
      } catch {
      }
    }
    return "";
  }
  function Ze(r, u) {
    return r === u || r !== r && u !== u;
  }
  var ut = Zt(function() {
    return arguments;
  }()) ? Zt : function(r) {
    return He(r) && we.call(r, "callee") && !Mr.call(r, "callee");
  }, lt = Array.isArray;
  function ft(r) {
    return r != null && ir(r.length) && !ct(r);
  }
  function qn(r) {
    return He(r) && ft(r);
  }
  var nr = Cr || Wn;
  function ct(r) {
    if (!Ne(r))
      return !1;
    var u = Je(r);
    return u == A || u == R || u == o || u == O;
  }
  function ir(r) {
    return typeof r == "number" && r > -1 && r % 1 == 0 && r <= p;
  }
  function Ne(r) {
    var u = typeof r;
    return r != null && (u == "object" || u == "function");
  }
  function He(r) {
    return r != null && typeof r == "object";
  }
  function Mn(r) {
    if (!He(r) || Je(r) != E)
      return !1;
    var u = Vt(r);
    if (u === null)
      return !0;
    var c = we.call(u, "constructor") && u.constructor;
    return typeof c == "function" && c instanceof c && _e.call(c) == rt;
  }
  var or = ee ? ie(ee) : dn;
  function Nn(r) {
    return yn(r, ar(r));
  }
  function ar(r) {
    return ft(r) ? ln(r, !0) : hn(r);
  }
  var Cn = Sn(function(r, u, c) {
    er(r, u, c);
  });
  function zn(r) {
    return function() {
      return r;
    };
  }
  function sr(r) {
    return r;
  }
  function Wn() {
    return !1;
  }
  a.exports = Cn;
})(et, et.exports);
var jn = et.exports;
function Qn() {
  if (!arguments.length)
    return [];
  var a = arguments[0];
  return Yn(a) ? a : [a];
}
var Yn = Array.isArray, Vn = Qn, dr = {}, hr = {}, Wt = { exports: {} }, ne = String, gr = function() {
  return { isColorSupported: !1, reset: ne, bold: ne, dim: ne, italic: ne, underline: ne, inverse: ne, hidden: ne, strikethrough: ne, black: ne, red: ne, green: ne, yellow: ne, blue: ne, magenta: ne, cyan: ne, white: ne, gray: ne, bgBlack: ne, bgRed: ne, bgGreen: ne, bgYellow: ne, bgBlue: ne, bgMagenta: ne, bgCyan: ne, bgWhite: ne };
};
Wt.exports = gr();
Wt.exports.createColors = gr;
var Kn = Wt.exports;
(function(a) {
  Object.defineProperty(a, "__esModule", {
    value: !0
  });
  function t(n, o) {
    for (var l in o)
      Object.defineProperty(n, l, {
        enumerable: !0,
        get: o[l]
      });
  }
  t(a, {
    dim: function() {
      return p;
    },
    default: function() {
      return m;
    }
  });
  const b = /* @__PURE__ */ d(Kn);
  function d(n) {
    return n && n.__esModule ? n : {
      default: n
    };
  }
  let g = /* @__PURE__ */ new Set();
  function y(n, o, l) {
    typeof process < "u" && process.env.JEST_WORKER_ID || l && g.has(l) || (l && g.add(l), console.warn(""), o.forEach((f) => console.warn(n, "-", f)));
  }
  function p(n) {
    return b.default.dim(n);
  }
  const m = {
    info(n, o) {
      y(b.default.bold(b.default.cyan("info")), ...Array.isArray(n) ? [
        n
      ] : [
        o,
        n
      ]);
    },
    warn(n, o) {
      y(b.default.bold(b.default.yellow("warn")), ...Array.isArray(n) ? [
        n
      ] : [
        o,
        n
      ]);
    },
    risk(n, o) {
      y(b.default.bold(b.default.magenta("risk")), ...Array.isArray(n) ? [
        n
      ] : [
        o,
        n
      ]);
    }
  };
})(hr);
(function(a) {
  Object.defineProperty(a, "__esModule", {
    value: !0
  }), Object.defineProperty(a, "default", {
    enumerable: !0,
    get: function() {
      return g;
    }
  });
  const t = /* @__PURE__ */ b(hr);
  function b(y) {
    return y && y.__esModule ? y : {
      default: y
    };
  }
  function d({ version: y, from: p, to: m }) {
    t.default.warn(`${p}-color-renamed`, [
      `As of Tailwind CSS ${y}, \`${p}\` has been renamed to \`${m}\`.`,
      "Update your configuration file to silence this warning."
    ]);
  }
  const g = {
    inherit: "inherit",
    current: "currentColor",
    transparent: "transparent",
    black: "#000",
    white: "#fff",
    slate: {
      50: "#f8fafc",
      100: "#f1f5f9",
      200: "#e2e8f0",
      300: "#cbd5e1",
      400: "#94a3b8",
      500: "#64748b",
      600: "#475569",
      700: "#334155",
      800: "#1e293b",
      900: "#0f172a",
      950: "#020617"
    },
    gray: {
      50: "#f9fafb",
      100: "#f3f4f6",
      200: "#e5e7eb",
      300: "#d1d5db",
      400: "#9ca3af",
      500: "#6b7280",
      600: "#4b5563",
      700: "#374151",
      800: "#1f2937",
      900: "#111827",
      950: "#030712"
    },
    zinc: {
      50: "#fafafa",
      100: "#f4f4f5",
      200: "#e4e4e7",
      300: "#d4d4d8",
      400: "#a1a1aa",
      500: "#71717a",
      600: "#52525b",
      700: "#3f3f46",
      800: "#27272a",
      900: "#18181b",
      950: "#09090b"
    },
    neutral: {
      50: "#fafafa",
      100: "#f5f5f5",
      200: "#e5e5e5",
      300: "#d4d4d4",
      400: "#a3a3a3",
      500: "#737373",
      600: "#525252",
      700: "#404040",
      800: "#262626",
      900: "#171717",
      950: "#0a0a0a"
    },
    stone: {
      50: "#fafaf9",
      100: "#f5f5f4",
      200: "#e7e5e4",
      300: "#d6d3d1",
      400: "#a8a29e",
      500: "#78716c",
      600: "#57534e",
      700: "#44403c",
      800: "#292524",
      900: "#1c1917",
      950: "#0c0a09"
    },
    red: {
      50: "#fef2f2",
      100: "#fee2e2",
      200: "#fecaca",
      300: "#fca5a5",
      400: "#f87171",
      500: "#ef4444",
      600: "#dc2626",
      700: "#b91c1c",
      800: "#991b1b",
      900: "#7f1d1d",
      950: "#450a0a"
    },
    orange: {
      50: "#fff7ed",
      100: "#ffedd5",
      200: "#fed7aa",
      300: "#fdba74",
      400: "#fb923c",
      500: "#f97316",
      600: "#ea580c",
      700: "#c2410c",
      800: "#9a3412",
      900: "#7c2d12",
      950: "#431407"
    },
    amber: {
      50: "#fffbeb",
      100: "#fef3c7",
      200: "#fde68a",
      300: "#fcd34d",
      400: "#fbbf24",
      500: "#f59e0b",
      600: "#d97706",
      700: "#b45309",
      800: "#92400e",
      900: "#78350f",
      950: "#451a03"
    },
    yellow: {
      50: "#fefce8",
      100: "#fef9c3",
      200: "#fef08a",
      300: "#fde047",
      400: "#facc15",
      500: "#eab308",
      600: "#ca8a04",
      700: "#a16207",
      800: "#854d0e",
      900: "#713f12",
      950: "#422006"
    },
    lime: {
      50: "#f7fee7",
      100: "#ecfccb",
      200: "#d9f99d",
      300: "#bef264",
      400: "#a3e635",
      500: "#84cc16",
      600: "#65a30d",
      700: "#4d7c0f",
      800: "#3f6212",
      900: "#365314",
      950: "#1a2e05"
    },
    green: {
      50: "#f0fdf4",
      100: "#dcfce7",
      200: "#bbf7d0",
      300: "#86efac",
      400: "#4ade80",
      500: "#22c55e",
      600: "#16a34a",
      700: "#15803d",
      800: "#166534",
      900: "#14532d",
      950: "#052e16"
    },
    emerald: {
      50: "#ecfdf5",
      100: "#d1fae5",
      200: "#a7f3d0",
      300: "#6ee7b7",
      400: "#34d399",
      500: "#10b981",
      600: "#059669",
      700: "#047857",
      800: "#065f46",
      900: "#064e3b",
      950: "#022c22"
    },
    teal: {
      50: "#f0fdfa",
      100: "#ccfbf1",
      200: "#99f6e4",
      300: "#5eead4",
      400: "#2dd4bf",
      500: "#14b8a6",
      600: "#0d9488",
      700: "#0f766e",
      800: "#115e59",
      900: "#134e4a",
      950: "#042f2e"
    },
    cyan: {
      50: "#ecfeff",
      100: "#cffafe",
      200: "#a5f3fc",
      300: "#67e8f9",
      400: "#22d3ee",
      500: "#06b6d4",
      600: "#0891b2",
      700: "#0e7490",
      800: "#155e75",
      900: "#164e63",
      950: "#083344"
    },
    sky: {
      50: "#f0f9ff",
      100: "#e0f2fe",
      200: "#bae6fd",
      300: "#7dd3fc",
      400: "#38bdf8",
      500: "#0ea5e9",
      600: "#0284c7",
      700: "#0369a1",
      800: "#075985",
      900: "#0c4a6e",
      950: "#082f49"
    },
    blue: {
      50: "#eff6ff",
      100: "#dbeafe",
      200: "#bfdbfe",
      300: "#93c5fd",
      400: "#60a5fa",
      500: "#3b82f6",
      600: "#2563eb",
      700: "#1d4ed8",
      800: "#1e40af",
      900: "#1e3a8a",
      950: "#172554"
    },
    indigo: {
      50: "#eef2ff",
      100: "#e0e7ff",
      200: "#c7d2fe",
      300: "#a5b4fc",
      400: "#818cf8",
      500: "#6366f1",
      600: "#4f46e5",
      700: "#4338ca",
      800: "#3730a3",
      900: "#312e81",
      950: "#1e1b4b"
    },
    violet: {
      50: "#f5f3ff",
      100: "#ede9fe",
      200: "#ddd6fe",
      300: "#c4b5fd",
      400: "#a78bfa",
      500: "#8b5cf6",
      600: "#7c3aed",
      700: "#6d28d9",
      800: "#5b21b6",
      900: "#4c1d95",
      950: "#2e1065"
    },
    purple: {
      50: "#faf5ff",
      100: "#f3e8ff",
      200: "#e9d5ff",
      300: "#d8b4fe",
      400: "#c084fc",
      500: "#a855f7",
      600: "#9333ea",
      700: "#7e22ce",
      800: "#6b21a8",
      900: "#581c87",
      950: "#3b0764"
    },
    fuchsia: {
      50: "#fdf4ff",
      100: "#fae8ff",
      200: "#f5d0fe",
      300: "#f0abfc",
      400: "#e879f9",
      500: "#d946ef",
      600: "#c026d3",
      700: "#a21caf",
      800: "#86198f",
      900: "#701a75",
      950: "#4a044e"
    },
    pink: {
      50: "#fdf2f8",
      100: "#fce7f3",
      200: "#fbcfe8",
      300: "#f9a8d4",
      400: "#f472b6",
      500: "#ec4899",
      600: "#db2777",
      700: "#be185d",
      800: "#9d174d",
      900: "#831843",
      950: "#500724"
    },
    rose: {
      50: "#fff1f2",
      100: "#ffe4e6",
      200: "#fecdd3",
      300: "#fda4af",
      400: "#fb7185",
      500: "#f43f5e",
      600: "#e11d48",
      700: "#be123c",
      800: "#9f1239",
      900: "#881337",
      950: "#4c0519"
    },
    get lightBlue() {
      return d({
        version: "v2.2",
        from: "lightBlue",
        to: "sky"
      }), this.sky;
    },
    get warmGray() {
      return d({
        version: "v3.0",
        from: "warmGray",
        to: "stone"
      }), this.stone;
    },
    get trueGray() {
      return d({
        version: "v3.0",
        from: "trueGray",
        to: "neutral"
      }), this.neutral;
    },
    get coolGray() {
      return d({
        version: "v3.0",
        from: "coolGray",
        to: "gray"
      }), this.gray;
    },
    get blueGray() {
      return d({
        version: "v3.0",
        from: "blueGray",
        to: "slate"
      }), this.slate;
    }
  };
})(dr);
let gt = dr;
var Jn = (gt.__esModule ? gt : { default: gt }).default;
const s = Jn, U = (a) => a.toFixed(7).replace(/(\.[0-9]+?)0+$/, "$1").replace(/\.0$/, ""), me = (a) => `${U(a / 16)}rem`, e = (a, t) => `${U(a / t)}em`, Ie = (a) => {
  a = a.replace("#", ""), a = a.length === 3 ? a.replace(/./g, "$&$&") : a;
  const t = parseInt(a.substring(0, 2), 16), b = parseInt(a.substring(2, 4), 16), d = parseInt(a.substring(4, 6), 16);
  return `${t} ${b} ${d}`;
};
let vt = {
  sm: {
    css: [
      {
        fontSize: me(14),
        lineHeight: U(24 / 14),
        p: {
          marginTop: e(16, 14),
          marginBottom: e(16, 14)
        },
        '[class~="lead"]': {
          fontSize: e(18, 14),
          lineHeight: U(28 / 18),
          marginTop: e(16, 18),
          marginBottom: e(16, 18)
        },
        blockquote: {
          marginTop: e(24, 18),
          marginBottom: e(24, 18),
          paddingLeft: e(20, 18)
        },
        h1: {
          fontSize: e(30, 14),
          marginTop: "0",
          marginBottom: e(24, 30),
          lineHeight: U(36 / 30)
        },
        h2: {
          fontSize: e(20, 14),
          marginTop: e(32, 20),
          marginBottom: e(16, 20),
          lineHeight: U(28 / 20)
        },
        h3: {
          fontSize: e(18, 14),
          marginTop: e(28, 18),
          marginBottom: e(8, 18),
          lineHeight: U(28 / 18)
        },
        h4: {
          marginTop: e(20, 14),
          marginBottom: e(8, 14),
          lineHeight: U(20 / 14)
        },
        img: {
          marginTop: e(24, 14),
          marginBottom: e(24, 14)
        },
        picture: {
          marginTop: e(24, 14),
          marginBottom: e(24, 14)
        },
        "picture > img": {
          marginTop: "0",
          marginBottom: "0"
        },
        video: {
          marginTop: e(24, 14),
          marginBottom: e(24, 14)
        },
        kbd: {
          fontSize: e(12, 14),
          borderRadius: me(5),
          paddingTop: e(2, 14),
          paddingRight: e(5, 14),
          paddingBottom: e(2, 14),
          paddingLeft: e(5, 14)
        },
        code: {
          fontSize: e(12, 14)
        },
        "h2 code": {
          fontSize: e(18, 20)
        },
        "h3 code": {
          fontSize: e(16, 18)
        },
        pre: {
          fontSize: e(12, 14),
          lineHeight: U(20 / 12),
          marginTop: e(20, 12),
          marginBottom: e(20, 12),
          borderRadius: me(4),
          paddingTop: e(8, 12),
          paddingRight: e(12, 12),
          paddingBottom: e(8, 12),
          paddingLeft: e(12, 12)
        },
        ol: {
          marginTop: e(16, 14),
          marginBottom: e(16, 14),
          paddingLeft: e(22, 14)
        },
        ul: {
          marginTop: e(16, 14),
          marginBottom: e(16, 14),
          paddingLeft: e(22, 14)
        },
        li: {
          marginTop: e(4, 14),
          marginBottom: e(4, 14)
        },
        "ol > li": {
          paddingLeft: e(6, 14)
        },
        "ul > li": {
          paddingLeft: e(6, 14)
        },
        "> ul > li p": {
          marginTop: e(8, 14),
          marginBottom: e(8, 14)
        },
        "> ul > li > *:first-child": {
          marginTop: e(16, 14)
        },
        "> ul > li > *:last-child": {
          marginBottom: e(16, 14)
        },
        "> ol > li > *:first-child": {
          marginTop: e(16, 14)
        },
        "> ol > li > *:last-child": {
          marginBottom: e(16, 14)
        },
        "ul ul, ul ol, ol ul, ol ol": {
          marginTop: e(8, 14),
          marginBottom: e(8, 14)
        },
        dl: {
          marginTop: e(16, 14),
          marginBottom: e(16, 14)
        },
        dt: {
          marginTop: e(16, 14)
        },
        dd: {
          marginTop: e(4, 14),
          paddingLeft: e(22, 14)
        },
        hr: {
          marginTop: e(40, 14),
          marginBottom: e(40, 14)
        },
        "hr + *": {
          marginTop: "0"
        },
        "h2 + *": {
          marginTop: "0"
        },
        "h3 + *": {
          marginTop: "0"
        },
        "h4 + *": {
          marginTop: "0"
        },
        table: {
          fontSize: e(12, 14),
          lineHeight: U(18 / 12)
        },
        "thead th": {
          paddingRight: e(12, 12),
          paddingBottom: e(8, 12),
          paddingLeft: e(12, 12)
        },
        "thead th:first-child": {
          paddingLeft: "0"
        },
        "thead th:last-child": {
          paddingRight: "0"
        },
        "tbody td, tfoot td": {
          paddingTop: e(8, 12),
          paddingRight: e(12, 12),
          paddingBottom: e(8, 12),
          paddingLeft: e(12, 12)
        },
        "tbody td:first-child, tfoot td:first-child": {
          paddingLeft: "0"
        },
        "tbody td:last-child, tfoot td:last-child": {
          paddingRight: "0"
        },
        figure: {
          marginTop: e(24, 14),
          marginBottom: e(24, 14)
        },
        "figure > *": {
          marginTop: "0",
          marginBottom: "0"
        },
        figcaption: {
          fontSize: e(12, 14),
          lineHeight: U(16 / 12),
          marginTop: e(8, 12)
        }
      },
      {
        "> :first-child": {
          marginTop: "0"
        },
        "> :last-child": {
          marginBottom: "0"
        }
      }
    ]
  },
  base: {
    css: [
      {
        fontSize: me(16),
        lineHeight: U(28 / 16),
        p: {
          marginTop: e(20, 16),
          marginBottom: e(20, 16)
        },
        '[class~="lead"]': {
          fontSize: e(20, 16),
          lineHeight: U(32 / 20),
          marginTop: e(24, 20),
          marginBottom: e(24, 20)
        },
        blockquote: {
          marginTop: e(32, 20),
          marginBottom: e(32, 20),
          paddingLeft: e(20, 20)
        },
        h1: {
          fontSize: e(36, 16),
          marginTop: "0",
          marginBottom: e(32, 36),
          lineHeight: U(40 / 36)
        },
        h2: {
          fontSize: e(24, 16),
          marginTop: e(48, 24),
          marginBottom: e(24, 24),
          lineHeight: U(32 / 24)
        },
        h3: {
          fontSize: e(20, 16),
          marginTop: e(32, 20),
          marginBottom: e(12, 20),
          lineHeight: U(32 / 20)
        },
        h4: {
          marginTop: e(24, 16),
          marginBottom: e(8, 16),
          lineHeight: U(24 / 16)
        },
        img: {
          marginTop: e(32, 16),
          marginBottom: e(32, 16)
        },
        picture: {
          marginTop: e(32, 16),
          marginBottom: e(32, 16)
        },
        "picture > img": {
          marginTop: "0",
          marginBottom: "0"
        },
        video: {
          marginTop: e(32, 16),
          marginBottom: e(32, 16)
        },
        kbd: {
          fontSize: e(14, 16),
          borderRadius: me(5),
          paddingTop: e(3, 16),
          paddingRight: e(6, 16),
          paddingBottom: e(3, 16),
          paddingLeft: e(6, 16)
        },
        code: {
          fontSize: e(14, 16)
        },
        "h2 code": {
          fontSize: e(21, 24)
        },
        "h3 code": {
          fontSize: e(18, 20)
        },
        pre: {
          fontSize: e(14, 16),
          lineHeight: U(24 / 14),
          marginTop: e(24, 14),
          marginBottom: e(24, 14),
          borderRadius: me(6),
          paddingTop: e(12, 14),
          paddingRight: e(16, 14),
          paddingBottom: e(12, 14),
          paddingLeft: e(16, 14)
        },
        ol: {
          marginTop: e(20, 16),
          marginBottom: e(20, 16),
          paddingLeft: e(26, 16)
        },
        ul: {
          marginTop: e(20, 16),
          marginBottom: e(20, 16),
          paddingLeft: e(26, 16)
        },
        li: {
          marginTop: e(8, 16),
          marginBottom: e(8, 16)
        },
        "ol > li": {
          paddingLeft: e(6, 16)
        },
        "ul > li": {
          paddingLeft: e(6, 16)
        },
        "> ul > li p": {
          marginTop: e(12, 16),
          marginBottom: e(12, 16)
        },
        "> ul > li > *:first-child": {
          marginTop: e(20, 16)
        },
        "> ul > li > *:last-child": {
          marginBottom: e(20, 16)
        },
        "> ol > li > *:first-child": {
          marginTop: e(20, 16)
        },
        "> ol > li > *:last-child": {
          marginBottom: e(20, 16)
        },
        "ul ul, ul ol, ol ul, ol ol": {
          marginTop: e(12, 16),
          marginBottom: e(12, 16)
        },
        dl: {
          marginTop: e(20, 16),
          marginBottom: e(20, 16)
        },
        dt: {
          marginTop: e(20, 16)
        },
        dd: {
          marginTop: e(8, 16),
          paddingLeft: e(26, 16)
        },
        hr: {
          marginTop: e(48, 16),
          marginBottom: e(48, 16)
        },
        "hr + *": {
          marginTop: "0"
        },
        "h2 + *": {
          marginTop: "0"
        },
        "h3 + *": {
          marginTop: "0"
        },
        "h4 + *": {
          marginTop: "0"
        },
        table: {
          fontSize: e(14, 16),
          lineHeight: U(24 / 14)
        },
        "thead th": {
          paddingRight: e(8, 14),
          paddingBottom: e(8, 14),
          paddingLeft: e(8, 14)
        },
        "thead th:first-child": {
          paddingLeft: "0"
        },
        "thead th:last-child": {
          paddingRight: "0"
        },
        "tbody td, tfoot td": {
          paddingTop: e(8, 14),
          paddingRight: e(8, 14),
          paddingBottom: e(8, 14),
          paddingLeft: e(8, 14)
        },
        "tbody td:first-child, tfoot td:first-child": {
          paddingLeft: "0"
        },
        "tbody td:last-child, tfoot td:last-child": {
          paddingRight: "0"
        },
        figure: {
          marginTop: e(32, 16),
          marginBottom: e(32, 16)
        },
        "figure > *": {
          marginTop: "0",
          marginBottom: "0"
        },
        figcaption: {
          fontSize: e(14, 16),
          lineHeight: U(20 / 14),
          marginTop: e(12, 14)
        }
      },
      {
        "> :first-child": {
          marginTop: "0"
        },
        "> :last-child": {
          marginBottom: "0"
        }
      }
    ]
  },
  lg: {
    css: [
      {
        fontSize: me(18),
        lineHeight: U(32 / 18),
        p: {
          marginTop: e(24, 18),
          marginBottom: e(24, 18)
        },
        '[class~="lead"]': {
          fontSize: e(22, 18),
          lineHeight: U(32 / 22),
          marginTop: e(24, 22),
          marginBottom: e(24, 22)
        },
        blockquote: {
          marginTop: e(40, 24),
          marginBottom: e(40, 24),
          paddingLeft: e(24, 24)
        },
        h1: {
          fontSize: e(48, 18),
          marginTop: "0",
          marginBottom: e(40, 48),
          lineHeight: U(48 / 48)
        },
        h2: {
          fontSize: e(30, 18),
          marginTop: e(56, 30),
          marginBottom: e(32, 30),
          lineHeight: U(40 / 30)
        },
        h3: {
          fontSize: e(24, 18),
          marginTop: e(40, 24),
          marginBottom: e(16, 24),
          lineHeight: U(36 / 24)
        },
        h4: {
          marginTop: e(32, 18),
          marginBottom: e(8, 18),
          lineHeight: U(28 / 18)
        },
        img: {
          marginTop: e(32, 18),
          marginBottom: e(32, 18)
        },
        picture: {
          marginTop: e(32, 18),
          marginBottom: e(32, 18)
        },
        "picture > img": {
          marginTop: "0",
          marginBottom: "0"
        },
        video: {
          marginTop: e(32, 18),
          marginBottom: e(32, 18)
        },
        kbd: {
          fontSize: e(16, 18),
          borderRadius: me(5),
          paddingTop: e(4, 18),
          paddingRight: e(8, 18),
          paddingBottom: e(4, 18),
          paddingLeft: e(8, 18)
        },
        code: {
          fontSize: e(16, 18)
        },
        "h2 code": {
          fontSize: e(26, 30)
        },
        "h3 code": {
          fontSize: e(21, 24)
        },
        pre: {
          fontSize: e(16, 18),
          lineHeight: U(28 / 16),
          marginTop: e(32, 16),
          marginBottom: e(32, 16),
          borderRadius: me(6),
          paddingTop: e(16, 16),
          paddingRight: e(24, 16),
          paddingBottom: e(16, 16),
          paddingLeft: e(24, 16)
        },
        ol: {
          marginTop: e(24, 18),
          marginBottom: e(24, 18),
          paddingLeft: e(28, 18)
        },
        ul: {
          marginTop: e(24, 18),
          marginBottom: e(24, 18),
          paddingLeft: e(28, 18)
        },
        li: {
          marginTop: e(12, 18),
          marginBottom: e(12, 18)
        },
        "ol > li": {
          paddingLeft: e(8, 18)
        },
        "ul > li": {
          paddingLeft: e(8, 18)
        },
        "> ul > li p": {
          marginTop: e(16, 18),
          marginBottom: e(16, 18)
        },
        "> ul > li > *:first-child": {
          marginTop: e(24, 18)
        },
        "> ul > li > *:last-child": {
          marginBottom: e(24, 18)
        },
        "> ol > li > *:first-child": {
          marginTop: e(24, 18)
        },
        "> ol > li > *:last-child": {
          marginBottom: e(24, 18)
        },
        "ul ul, ul ol, ol ul, ol ol": {
          marginTop: e(16, 18),
          marginBottom: e(16, 18)
        },
        dl: {
          marginTop: e(24, 18),
          marginBottom: e(24, 18)
        },
        dt: {
          marginTop: e(24, 18)
        },
        dd: {
          marginTop: e(12, 18),
          paddingLeft: e(28, 18)
        },
        hr: {
          marginTop: e(56, 18),
          marginBottom: e(56, 18)
        },
        "hr + *": {
          marginTop: "0"
        },
        "h2 + *": {
          marginTop: "0"
        },
        "h3 + *": {
          marginTop: "0"
        },
        "h4 + *": {
          marginTop: "0"
        },
        table: {
          fontSize: e(16, 18),
          lineHeight: U(24 / 16)
        },
        "thead th": {
          paddingRight: e(12, 16),
          paddingBottom: e(12, 16),
          paddingLeft: e(12, 16)
        },
        "thead th:first-child": {
          paddingLeft: "0"
        },
        "thead th:last-child": {
          paddingRight: "0"
        },
        "tbody td, tfoot td": {
          paddingTop: e(12, 16),
          paddingRight: e(12, 16),
          paddingBottom: e(12, 16),
          paddingLeft: e(12, 16)
        },
        "tbody td:first-child, tfoot td:first-child": {
          paddingLeft: "0"
        },
        "tbody td:last-child, tfoot td:last-child": {
          paddingRight: "0"
        },
        figure: {
          marginTop: e(32, 18),
          marginBottom: e(32, 18)
        },
        "figure > *": {
          marginTop: "0",
          marginBottom: "0"
        },
        figcaption: {
          fontSize: e(16, 18),
          lineHeight: U(24 / 16),
          marginTop: e(16, 16)
        }
      },
      {
        "> :first-child": {
          marginTop: "0"
        },
        "> :last-child": {
          marginBottom: "0"
        }
      }
    ]
  },
  xl: {
    css: [
      {
        fontSize: me(20),
        lineHeight: U(36 / 20),
        p: {
          marginTop: e(24, 20),
          marginBottom: e(24, 20)
        },
        '[class~="lead"]': {
          fontSize: e(24, 20),
          lineHeight: U(36 / 24),
          marginTop: e(24, 24),
          marginBottom: e(24, 24)
        },
        blockquote: {
          marginTop: e(48, 30),
          marginBottom: e(48, 30),
          paddingLeft: e(32, 30)
        },
        h1: {
          fontSize: e(56, 20),
          marginTop: "0",
          marginBottom: e(48, 56),
          lineHeight: U(56 / 56)
        },
        h2: {
          fontSize: e(36, 20),
          marginTop: e(56, 36),
          marginBottom: e(32, 36),
          lineHeight: U(40 / 36)
        },
        h3: {
          fontSize: e(30, 20),
          marginTop: e(48, 30),
          marginBottom: e(20, 30),
          lineHeight: U(40 / 30)
        },
        h4: {
          marginTop: e(36, 20),
          marginBottom: e(12, 20),
          lineHeight: U(32 / 20)
        },
        img: {
          marginTop: e(40, 20),
          marginBottom: e(40, 20)
        },
        picture: {
          marginTop: e(40, 20),
          marginBottom: e(40, 20)
        },
        "picture > img": {
          marginTop: "0",
          marginBottom: "0"
        },
        video: {
          marginTop: e(40, 20),
          marginBottom: e(40, 20)
        },
        kbd: {
          fontSize: e(18, 20),
          borderRadius: me(5),
          paddingTop: e(5, 20),
          paddingRight: e(8, 20),
          paddingBottom: e(5, 20),
          paddingLeft: e(8, 20)
        },
        code: {
          fontSize: e(18, 20)
        },
        "h2 code": {
          fontSize: e(31, 36)
        },
        "h3 code": {
          fontSize: e(27, 30)
        },
        pre: {
          fontSize: e(18, 20),
          lineHeight: U(32 / 18),
          marginTop: e(36, 18),
          marginBottom: e(36, 18),
          borderRadius: me(8),
          paddingTop: e(20, 18),
          paddingRight: e(24, 18),
          paddingBottom: e(20, 18),
          paddingLeft: e(24, 18)
        },
        ol: {
          marginTop: e(24, 20),
          marginBottom: e(24, 20),
          paddingLeft: e(32, 20)
        },
        ul: {
          marginTop: e(24, 20),
          marginBottom: e(24, 20),
          paddingLeft: e(32, 20)
        },
        li: {
          marginTop: e(12, 20),
          marginBottom: e(12, 20)
        },
        "ol > li": {
          paddingLeft: e(8, 20)
        },
        "ul > li": {
          paddingLeft: e(8, 20)
        },
        "> ul > li p": {
          marginTop: e(16, 20),
          marginBottom: e(16, 20)
        },
        "> ul > li > *:first-child": {
          marginTop: e(24, 20)
        },
        "> ul > li > *:last-child": {
          marginBottom: e(24, 20)
        },
        "> ol > li > *:first-child": {
          marginTop: e(24, 20)
        },
        "> ol > li > *:last-child": {
          marginBottom: e(24, 20)
        },
        "ul ul, ul ol, ol ul, ol ol": {
          marginTop: e(16, 20),
          marginBottom: e(16, 20)
        },
        dl: {
          marginTop: e(24, 20),
          marginBottom: e(24, 20)
        },
        dt: {
          marginTop: e(24, 20)
        },
        dd: {
          marginTop: e(12, 20),
          paddingLeft: e(32, 20)
        },
        hr: {
          marginTop: e(56, 20),
          marginBottom: e(56, 20)
        },
        "hr + *": {
          marginTop: "0"
        },
        "h2 + *": {
          marginTop: "0"
        },
        "h3 + *": {
          marginTop: "0"
        },
        "h4 + *": {
          marginTop: "0"
        },
        table: {
          fontSize: e(18, 20),
          lineHeight: U(28 / 18)
        },
        "thead th": {
          paddingRight: e(12, 18),
          paddingBottom: e(16, 18),
          paddingLeft: e(12, 18)
        },
        "thead th:first-child": {
          paddingLeft: "0"
        },
        "thead th:last-child": {
          paddingRight: "0"
        },
        "tbody td, tfoot td": {
          paddingTop: e(16, 18),
          paddingRight: e(12, 18),
          paddingBottom: e(16, 18),
          paddingLeft: e(12, 18)
        },
        "tbody td:first-child, tfoot td:first-child": {
          paddingLeft: "0"
        },
        "tbody td:last-child, tfoot td:last-child": {
          paddingRight: "0"
        },
        figure: {
          marginTop: e(40, 20),
          marginBottom: e(40, 20)
        },
        "figure > *": {
          marginTop: "0",
          marginBottom: "0"
        },
        figcaption: {
          fontSize: e(18, 20),
          lineHeight: U(28 / 18),
          marginTop: e(18, 18)
        }
      },
      {
        "> :first-child": {
          marginTop: "0"
        },
        "> :last-child": {
          marginBottom: "0"
        }
      }
    ]
  },
  "2xl": {
    css: [
      {
        fontSize: me(24),
        lineHeight: U(40 / 24),
        p: {
          marginTop: e(32, 24),
          marginBottom: e(32, 24)
        },
        '[class~="lead"]': {
          fontSize: e(30, 24),
          lineHeight: U(44 / 30),
          marginTop: e(32, 30),
          marginBottom: e(32, 30)
        },
        blockquote: {
          marginTop: e(64, 36),
          marginBottom: e(64, 36),
          paddingLeft: e(40, 36)
        },
        h1: {
          fontSize: e(64, 24),
          marginTop: "0",
          marginBottom: e(56, 64),
          lineHeight: U(64 / 64)
        },
        h2: {
          fontSize: e(48, 24),
          marginTop: e(72, 48),
          marginBottom: e(40, 48),
          lineHeight: U(52 / 48)
        },
        h3: {
          fontSize: e(36, 24),
          marginTop: e(56, 36),
          marginBottom: e(24, 36),
          lineHeight: U(44 / 36)
        },
        h4: {
          marginTop: e(40, 24),
          marginBottom: e(16, 24),
          lineHeight: U(36 / 24)
        },
        img: {
          marginTop: e(48, 24),
          marginBottom: e(48, 24)
        },
        picture: {
          marginTop: e(48, 24),
          marginBottom: e(48, 24)
        },
        "picture > img": {
          marginTop: "0",
          marginBottom: "0"
        },
        video: {
          marginTop: e(48, 24),
          marginBottom: e(48, 24)
        },
        kbd: {
          fontSize: e(20, 24),
          borderRadius: me(6),
          paddingTop: e(6, 24),
          paddingRight: e(8, 24),
          paddingBottom: e(6, 24),
          paddingLeft: e(8, 24)
        },
        code: {
          fontSize: e(20, 24)
        },
        "h2 code": {
          fontSize: e(42, 48)
        },
        "h3 code": {
          fontSize: e(32, 36)
        },
        pre: {
          fontSize: e(20, 24),
          lineHeight: U(36 / 20),
          marginTop: e(40, 20),
          marginBottom: e(40, 20),
          borderRadius: me(8),
          paddingTop: e(24, 20),
          paddingRight: e(32, 20),
          paddingBottom: e(24, 20),
          paddingLeft: e(32, 20)
        },
        ol: {
          marginTop: e(32, 24),
          marginBottom: e(32, 24),
          paddingLeft: e(38, 24)
        },
        ul: {
          marginTop: e(32, 24),
          marginBottom: e(32, 24),
          paddingLeft: e(38, 24)
        },
        li: {
          marginTop: e(12, 24),
          marginBottom: e(12, 24)
        },
        "ol > li": {
          paddingLeft: e(10, 24)
        },
        "ul > li": {
          paddingLeft: e(10, 24)
        },
        "> ul > li p": {
          marginTop: e(20, 24),
          marginBottom: e(20, 24)
        },
        "> ul > li > *:first-child": {
          marginTop: e(32, 24)
        },
        "> ul > li > *:last-child": {
          marginBottom: e(32, 24)
        },
        "> ol > li > *:first-child": {
          marginTop: e(32, 24)
        },
        "> ol > li > *:last-child": {
          marginBottom: e(32, 24)
        },
        "ul ul, ul ol, ol ul, ol ol": {
          marginTop: e(16, 24),
          marginBottom: e(16, 24)
        },
        dl: {
          marginTop: e(32, 24),
          marginBottom: e(32, 24)
        },
        dt: {
          marginTop: e(32, 24)
        },
        dd: {
          marginTop: e(12, 24),
          paddingLeft: e(38, 24)
        },
        hr: {
          marginTop: e(72, 24),
          marginBottom: e(72, 24)
        },
        "hr + *": {
          marginTop: "0"
        },
        "h2 + *": {
          marginTop: "0"
        },
        "h3 + *": {
          marginTop: "0"
        },
        "h4 + *": {
          marginTop: "0"
        },
        table: {
          fontSize: e(20, 24),
          lineHeight: U(28 / 20)
        },
        "thead th": {
          paddingRight: e(12, 20),
          paddingBottom: e(16, 20),
          paddingLeft: e(12, 20)
        },
        "thead th:first-child": {
          paddingLeft: "0"
        },
        "thead th:last-child": {
          paddingRight: "0"
        },
        "tbody td, tfoot td": {
          paddingTop: e(16, 20),
          paddingRight: e(12, 20),
          paddingBottom: e(16, 20),
          paddingLeft: e(12, 20)
        },
        "tbody td:first-child, tfoot td:first-child": {
          paddingLeft: "0"
        },
        "tbody td:last-child, tfoot td:last-child": {
          paddingRight: "0"
        },
        figure: {
          marginTop: e(48, 24),
          marginBottom: e(48, 24)
        },
        "figure > *": {
          marginTop: "0",
          marginBottom: "0"
        },
        figcaption: {
          fontSize: e(20, 24),
          lineHeight: U(32 / 20),
          marginTop: e(20, 20)
        }
      },
      {
        "> :first-child": {
          marginTop: "0"
        },
        "> :last-child": {
          marginBottom: "0"
        }
      }
    ]
  },
  // Gray color themes
  slate: {
    css: {
      "--tw-prose-body": s.slate[700],
      "--tw-prose-headings": s.slate[900],
      "--tw-prose-lead": s.slate[600],
      "--tw-prose-links": s.slate[900],
      "--tw-prose-bold": s.slate[900],
      "--tw-prose-counters": s.slate[500],
      "--tw-prose-bullets": s.slate[300],
      "--tw-prose-hr": s.slate[200],
      "--tw-prose-quotes": s.slate[900],
      "--tw-prose-quote-borders": s.slate[200],
      "--tw-prose-captions": s.slate[500],
      "--tw-prose-kbd": s.slate[900],
      "--tw-prose-kbd-shadows": Ie(s.slate[900]),
      "--tw-prose-code": s.slate[900],
      "--tw-prose-pre-code": s.slate[200],
      "--tw-prose-pre-bg": s.slate[800],
      "--tw-prose-th-borders": s.slate[300],
      "--tw-prose-td-borders": s.slate[200],
      "--tw-prose-invert-body": s.slate[300],
      "--tw-prose-invert-headings": s.white,
      "--tw-prose-invert-lead": s.slate[400],
      "--tw-prose-invert-links": s.white,
      "--tw-prose-invert-bold": s.white,
      "--tw-prose-invert-counters": s.slate[400],
      "--tw-prose-invert-bullets": s.slate[600],
      "--tw-prose-invert-hr": s.slate[700],
      "--tw-prose-invert-quotes": s.slate[100],
      "--tw-prose-invert-quote-borders": s.slate[700],
      "--tw-prose-invert-captions": s.slate[400],
      "--tw-prose-invert-kbd": s.white,
      "--tw-prose-invert-kbd-shadows": Ie(s.white),
      "--tw-prose-invert-code": s.white,
      "--tw-prose-invert-pre-code": s.slate[300],
      "--tw-prose-invert-pre-bg": "rgb(0 0 0 / 50%)",
      "--tw-prose-invert-th-borders": s.slate[600],
      "--tw-prose-invert-td-borders": s.slate[700]
    }
  },
  gray: {
    css: {
      "--tw-prose-body": s.gray[700],
      "--tw-prose-headings": s.gray[900],
      "--tw-prose-lead": s.gray[600],
      "--tw-prose-links": s.gray[900],
      "--tw-prose-bold": s.gray[900],
      "--tw-prose-counters": s.gray[500],
      "--tw-prose-bullets": s.gray[300],
      "--tw-prose-hr": s.gray[200],
      "--tw-prose-quotes": s.gray[900],
      "--tw-prose-quote-borders": s.gray[200],
      "--tw-prose-captions": s.gray[500],
      "--tw-prose-kbd": s.gray[900],
      "--tw-prose-kbd-shadows": Ie(s.gray[900]),
      "--tw-prose-code": s.gray[900],
      "--tw-prose-pre-code": s.gray[200],
      "--tw-prose-pre-bg": s.gray[800],
      "--tw-prose-th-borders": s.gray[300],
      "--tw-prose-td-borders": s.gray[200],
      "--tw-prose-invert-body": s.gray[300],
      "--tw-prose-invert-headings": s.white,
      "--tw-prose-invert-lead": s.gray[400],
      "--tw-prose-invert-links": s.white,
      "--tw-prose-invert-bold": s.white,
      "--tw-prose-invert-counters": s.gray[400],
      "--tw-prose-invert-bullets": s.gray[600],
      "--tw-prose-invert-hr": s.gray[700],
      "--tw-prose-invert-quotes": s.gray[100],
      "--tw-prose-invert-quote-borders": s.gray[700],
      "--tw-prose-invert-captions": s.gray[400],
      "--tw-prose-invert-kbd": s.white,
      "--tw-prose-invert-kbd-shadows": Ie(s.white),
      "--tw-prose-invert-code": s.white,
      "--tw-prose-invert-pre-code": s.gray[300],
      "--tw-prose-invert-pre-bg": "rgb(0 0 0 / 50%)",
      "--tw-prose-invert-th-borders": s.gray[600],
      "--tw-prose-invert-td-borders": s.gray[700]
    }
  },
  zinc: {
    css: {
      "--tw-prose-body": s.zinc[700],
      "--tw-prose-headings": s.zinc[900],
      "--tw-prose-lead": s.zinc[600],
      "--tw-prose-links": s.zinc[900],
      "--tw-prose-bold": s.zinc[900],
      "--tw-prose-counters": s.zinc[500],
      "--tw-prose-bullets": s.zinc[300],
      "--tw-prose-hr": s.zinc[200],
      "--tw-prose-quotes": s.zinc[900],
      "--tw-prose-quote-borders": s.zinc[200],
      "--tw-prose-captions": s.zinc[500],
      "--tw-prose-kbd": s.zinc[900],
      "--tw-prose-kbd-shadows": Ie(s.zinc[900]),
      "--tw-prose-code": s.zinc[900],
      "--tw-prose-pre-code": s.zinc[200],
      "--tw-prose-pre-bg": s.zinc[800],
      "--tw-prose-th-borders": s.zinc[300],
      "--tw-prose-td-borders": s.zinc[200],
      "--tw-prose-invert-body": s.zinc[300],
      "--tw-prose-invert-headings": s.white,
      "--tw-prose-invert-lead": s.zinc[400],
      "--tw-prose-invert-links": s.white,
      "--tw-prose-invert-bold": s.white,
      "--tw-prose-invert-counters": s.zinc[400],
      "--tw-prose-invert-bullets": s.zinc[600],
      "--tw-prose-invert-hr": s.zinc[700],
      "--tw-prose-invert-quotes": s.zinc[100],
      "--tw-prose-invert-quote-borders": s.zinc[700],
      "--tw-prose-invert-captions": s.zinc[400],
      "--tw-prose-invert-kbd": s.white,
      "--tw-prose-invert-kbd-shadows": Ie(s.white),
      "--tw-prose-invert-code": s.white,
      "--tw-prose-invert-pre-code": s.zinc[300],
      "--tw-prose-invert-pre-bg": "rgb(0 0 0 / 50%)",
      "--tw-prose-invert-th-borders": s.zinc[600],
      "--tw-prose-invert-td-borders": s.zinc[700]
    }
  },
  neutral: {
    css: {
      "--tw-prose-body": s.neutral[700],
      "--tw-prose-headings": s.neutral[900],
      "--tw-prose-lead": s.neutral[600],
      "--tw-prose-links": s.neutral[900],
      "--tw-prose-bold": s.neutral[900],
      "--tw-prose-counters": s.neutral[500],
      "--tw-prose-bullets": s.neutral[300],
      "--tw-prose-hr": s.neutral[200],
      "--tw-prose-quotes": s.neutral[900],
      "--tw-prose-quote-borders": s.neutral[200],
      "--tw-prose-captions": s.neutral[500],
      "--tw-prose-kbd": s.neutral[900],
      "--tw-prose-kbd-shadows": Ie(s.neutral[900]),
      "--tw-prose-code": s.neutral[900],
      "--tw-prose-pre-code": s.neutral[200],
      "--tw-prose-pre-bg": s.neutral[800],
      "--tw-prose-th-borders": s.neutral[300],
      "--tw-prose-td-borders": s.neutral[200],
      "--tw-prose-invert-body": s.neutral[300],
      "--tw-prose-invert-headings": s.white,
      "--tw-prose-invert-lead": s.neutral[400],
      "--tw-prose-invert-links": s.white,
      "--tw-prose-invert-bold": s.white,
      "--tw-prose-invert-counters": s.neutral[400],
      "--tw-prose-invert-bullets": s.neutral[600],
      "--tw-prose-invert-hr": s.neutral[700],
      "--tw-prose-invert-quotes": s.neutral[100],
      "--tw-prose-invert-quote-borders": s.neutral[700],
      "--tw-prose-invert-captions": s.neutral[400],
      "--tw-prose-invert-kbd": s.white,
      "--tw-prose-invert-kbd-shadows": Ie(s.white),
      "--tw-prose-invert-code": s.white,
      "--tw-prose-invert-pre-code": s.neutral[300],
      "--tw-prose-invert-pre-bg": "rgb(0 0 0 / 50%)",
      "--tw-prose-invert-th-borders": s.neutral[600],
      "--tw-prose-invert-td-borders": s.neutral[700]
    }
  },
  stone: {
    css: {
      "--tw-prose-body": s.stone[700],
      "--tw-prose-headings": s.stone[900],
      "--tw-prose-lead": s.stone[600],
      "--tw-prose-links": s.stone[900],
      "--tw-prose-bold": s.stone[900],
      "--tw-prose-counters": s.stone[500],
      "--tw-prose-bullets": s.stone[300],
      "--tw-prose-hr": s.stone[200],
      "--tw-prose-quotes": s.stone[900],
      "--tw-prose-quote-borders": s.stone[200],
      "--tw-prose-captions": s.stone[500],
      "--tw-prose-kbd": s.stone[900],
      "--tw-prose-kbd-shadows": Ie(s.stone[900]),
      "--tw-prose-code": s.stone[900],
      "--tw-prose-pre-code": s.stone[200],
      "--tw-prose-pre-bg": s.stone[800],
      "--tw-prose-th-borders": s.stone[300],
      "--tw-prose-td-borders": s.stone[200],
      "--tw-prose-invert-body": s.stone[300],
      "--tw-prose-invert-headings": s.white,
      "--tw-prose-invert-lead": s.stone[400],
      "--tw-prose-invert-links": s.white,
      "--tw-prose-invert-bold": s.white,
      "--tw-prose-invert-counters": s.stone[400],
      "--tw-prose-invert-bullets": s.stone[600],
      "--tw-prose-invert-hr": s.stone[700],
      "--tw-prose-invert-quotes": s.stone[100],
      "--tw-prose-invert-quote-borders": s.stone[700],
      "--tw-prose-invert-captions": s.stone[400],
      "--tw-prose-invert-kbd": s.white,
      "--tw-prose-invert-kbd-shadows": Ie(s.white),
      "--tw-prose-invert-code": s.white,
      "--tw-prose-invert-pre-code": s.stone[300],
      "--tw-prose-invert-pre-bg": "rgb(0 0 0 / 50%)",
      "--tw-prose-invert-th-borders": s.stone[600],
      "--tw-prose-invert-td-borders": s.stone[700]
    }
  },
  // Link-only themes (for backward compatibility)
  red: {
    css: {
      "--tw-prose-links": s.red[600],
      "--tw-prose-invert-links": s.red[500]
    }
  },
  orange: {
    css: {
      "--tw-prose-links": s.orange[600],
      "--tw-prose-invert-links": s.orange[500]
    }
  },
  amber: {
    css: {
      "--tw-prose-links": s.amber[600],
      "--tw-prose-invert-links": s.amber[500]
    }
  },
  yellow: {
    css: {
      "--tw-prose-links": s.yellow[600],
      "--tw-prose-invert-links": s.yellow[500]
    }
  },
  lime: {
    css: {
      "--tw-prose-links": s.lime[600],
      "--tw-prose-invert-links": s.lime[500]
    }
  },
  green: {
    css: {
      "--tw-prose-links": s.green[600],
      "--tw-prose-invert-links": s.green[500]
    }
  },
  emerald: {
    css: {
      "--tw-prose-links": s.emerald[600],
      "--tw-prose-invert-links": s.emerald[500]
    }
  },
  teal: {
    css: {
      "--tw-prose-links": s.teal[600],
      "--tw-prose-invert-links": s.teal[500]
    }
  },
  cyan: {
    css: {
      "--tw-prose-links": s.cyan[600],
      "--tw-prose-invert-links": s.cyan[500]
    }
  },
  sky: {
    css: {
      "--tw-prose-links": s.sky[600],
      "--tw-prose-invert-links": s.sky[500]
    }
  },
  blue: {
    css: {
      "--tw-prose-links": s.blue[600],
      "--tw-prose-invert-links": s.blue[500]
    }
  },
  indigo: {
    css: {
      "--tw-prose-links": s.indigo[600],
      "--tw-prose-invert-links": s.indigo[500]
    }
  },
  violet: {
    css: {
      "--tw-prose-links": s.violet[600],
      "--tw-prose-invert-links": s.violet[500]
    }
  },
  purple: {
    css: {
      "--tw-prose-links": s.purple[600],
      "--tw-prose-invert-links": s.purple[500]
    }
  },
  fuchsia: {
    css: {
      "--tw-prose-links": s.fuchsia[600],
      "--tw-prose-invert-links": s.fuchsia[500]
    }
  },
  pink: {
    css: {
      "--tw-prose-links": s.pink[600],
      "--tw-prose-invert-links": s.pink[500]
    }
  },
  rose: {
    css: {
      "--tw-prose-links": s.rose[600],
      "--tw-prose-invert-links": s.rose[500]
    }
  },
  // Invert (for dark mode)
  invert: {
    css: {
      "--tw-prose-body": "var(--tw-prose-invert-body)",
      "--tw-prose-headings": "var(--tw-prose-invert-headings)",
      "--tw-prose-lead": "var(--tw-prose-invert-lead)",
      "--tw-prose-links": "var(--tw-prose-invert-links)",
      "--tw-prose-bold": "var(--tw-prose-invert-bold)",
      "--tw-prose-counters": "var(--tw-prose-invert-counters)",
      "--tw-prose-bullets": "var(--tw-prose-invert-bullets)",
      "--tw-prose-hr": "var(--tw-prose-invert-hr)",
      "--tw-prose-quotes": "var(--tw-prose-invert-quotes)",
      "--tw-prose-quote-borders": "var(--tw-prose-invert-quote-borders)",
      "--tw-prose-captions": "var(--tw-prose-invert-captions)",
      "--tw-prose-kbd": "var(--tw-prose-invert-kbd)",
      "--tw-prose-kbd-shadows": "var(--tw-prose-invert-kbd-shadows)",
      "--tw-prose-code": "var(--tw-prose-invert-code)",
      "--tw-prose-pre-code": "var(--tw-prose-invert-pre-code)",
      "--tw-prose-pre-bg": "var(--tw-prose-invert-pre-bg)",
      "--tw-prose-th-borders": "var(--tw-prose-invert-th-borders)",
      "--tw-prose-td-borders": "var(--tw-prose-invert-td-borders)"
    }
  }
};
var Xn = {
  DEFAULT: {
    css: [
      {
        color: "var(--tw-prose-body)",
        maxWidth: "65ch",
        p: {},
        // Required to maintain correct order when merging
        '[class~="lead"]': {
          color: "var(--tw-prose-lead)"
        },
        a: {
          color: "var(--tw-prose-links)",
          textDecoration: "underline",
          fontWeight: "500"
        },
        strong: {
          color: "var(--tw-prose-bold)",
          fontWeight: "600"
        },
        "a strong": {
          color: "inherit"
        },
        "blockquote strong": {
          color: "inherit"
        },
        "thead th strong": {
          color: "inherit"
        },
        ol: {
          listStyleType: "decimal"
        },
        'ol[type="A"]': {
          listStyleType: "upper-alpha"
        },
        'ol[type="a"]': {
          listStyleType: "lower-alpha"
        },
        'ol[type="A" s]': {
          listStyleType: "upper-alpha"
        },
        'ol[type="a" s]': {
          listStyleType: "lower-alpha"
        },
        'ol[type="I"]': {
          listStyleType: "upper-roman"
        },
        'ol[type="i"]': {
          listStyleType: "lower-roman"
        },
        'ol[type="I" s]': {
          listStyleType: "upper-roman"
        },
        'ol[type="i" s]': {
          listStyleType: "lower-roman"
        },
        'ol[type="1"]': {
          listStyleType: "decimal"
        },
        ul: {
          listStyleType: "disc"
        },
        "ol > li::marker": {
          fontWeight: "400",
          color: "var(--tw-prose-counters)"
        },
        "ul > li::marker": {
          color: "var(--tw-prose-bullets)"
        },
        dt: {
          color: "var(--tw-prose-headings)",
          fontWeight: "600"
        },
        hr: {
          borderColor: "var(--tw-prose-hr)",
          borderTopWidth: 1
        },
        blockquote: {
          fontWeight: "500",
          fontStyle: "italic",
          color: "var(--tw-prose-quotes)",
          borderLeftWidth: "0.25rem",
          borderLeftColor: "var(--tw-prose-quote-borders)",
          quotes: '"\\201C""\\201D""\\2018""\\2019"'
        },
        "blockquote p:first-of-type::before": {
          content: "open-quote"
        },
        "blockquote p:last-of-type::after": {
          content: "close-quote"
        },
        h1: {
          color: "var(--tw-prose-headings)",
          fontWeight: "800"
        },
        "h1 strong": {
          fontWeight: "900",
          color: "inherit"
        },
        h2: {
          color: "var(--tw-prose-headings)",
          fontWeight: "700"
        },
        "h2 strong": {
          fontWeight: "800",
          color: "inherit"
        },
        h3: {
          color: "var(--tw-prose-headings)",
          fontWeight: "600"
        },
        "h3 strong": {
          fontWeight: "700",
          color: "inherit"
        },
        h4: {
          color: "var(--tw-prose-headings)",
          fontWeight: "600"
        },
        "h4 strong": {
          fontWeight: "700",
          color: "inherit"
        },
        img: {},
        // Required to maintain correct order when merging
        picture: {
          display: "block"
        },
        kbd: {
          fontWeight: "500",
          fontFamily: "inherit",
          color: "var(--tw-prose-kbd)",
          boxShadow: "0 0 0 1px rgb(var(--tw-prose-kbd-shadows) / 10%), 0 3px 0 rgb(var(--tw-prose-kbd-shadows) / 10%)"
        },
        code: {
          color: "var(--tw-prose-code)",
          fontWeight: "600"
        },
        "code::before": {
          content: '"`"'
        },
        "code::after": {
          content: '"`"'
        },
        "a code": {
          color: "inherit"
        },
        "h1 code": {
          color: "inherit"
        },
        "h2 code": {
          color: "inherit"
        },
        "h3 code": {
          color: "inherit"
        },
        "h4 code": {
          color: "inherit"
        },
        "blockquote code": {
          color: "inherit"
        },
        "thead th code": {
          color: "inherit"
        },
        pre: {
          color: "var(--tw-prose-pre-code)",
          backgroundColor: "var(--tw-prose-pre-bg)",
          overflowX: "auto",
          fontWeight: "400"
        },
        "pre code": {
          backgroundColor: "transparent",
          borderWidth: "0",
          borderRadius: "0",
          padding: "0",
          fontWeight: "inherit",
          color: "inherit",
          fontSize: "inherit",
          fontFamily: "inherit",
          lineHeight: "inherit"
        },
        "pre code::before": {
          content: "none"
        },
        "pre code::after": {
          content: "none"
        },
        table: {
          width: "100%",
          tableLayout: "auto",
          textAlign: "left",
          marginTop: e(32, 16),
          marginBottom: e(32, 16)
        },
        thead: {
          borderBottomWidth: "1px",
          borderBottomColor: "var(--tw-prose-th-borders)"
        },
        "thead th": {
          color: "var(--tw-prose-headings)",
          fontWeight: "600",
          verticalAlign: "bottom"
        },
        "tbody tr": {
          borderBottomWidth: "1px",
          borderBottomColor: "var(--tw-prose-td-borders)"
        },
        "tbody tr:last-child": {
          borderBottomWidth: "0"
        },
        "tbody td": {
          verticalAlign: "baseline"
        },
        tfoot: {
          borderTopWidth: "1px",
          borderTopColor: "var(--tw-prose-th-borders)"
        },
        "tfoot td": {
          verticalAlign: "top"
        },
        "figure > *": {},
        // Required to maintain correct order when merging
        figcaption: {
          color: "var(--tw-prose-captions)"
        }
      },
      vt.gray.css,
      ...vt.base.css
    ]
  },
  ...vt
}, Zn = "[object Object]";
function ei(a) {
  var t = !1;
  if (a != null && typeof a.toString != "function")
    try {
      t = !!(a + "");
    } catch {
    }
  return t;
}
function ti(a, t) {
  return function(b) {
    return a(t(b));
  };
}
var ri = Function.prototype, vr = Object.prototype, mr = ri.toString, ni = vr.hasOwnProperty, ii = mr.call(Object), oi = vr.toString, ai = ti(Object.getPrototypeOf, Object);
function si(a) {
  return !!a && typeof a == "object";
}
function ui(a) {
  if (!si(a) || oi.call(a) != Zn || ei(a))
    return !1;
  var t = ai(a);
  if (t === null)
    return !0;
  var b = ni.call(t, "constructor") && t.constructor;
  return typeof b == "function" && b instanceof b && mr.call(b) == ii;
}
var li = ui, wt = { exports: {} }, bt = { exports: {} }, _t = { exports: {} }, Tt = { exports: {} }, yt = { exports: {} }, St = { exports: {} }, ye = {}, Ot = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = g;
  function b(y) {
    for (var p = y.toLowerCase(), m = "", n = !1, o = 0; o < 6 && p[o] !== void 0; o++) {
      var l = p.charCodeAt(o), f = l >= 97 && l <= 102 || l >= 48 && l <= 57;
      if (n = l === 32, !f)
        break;
      m += p[o];
    }
    if (m.length !== 0) {
      var w = parseInt(m, 16), A = w >= 55296 && w <= 57343;
      return A || w === 0 || w > 1114111 ? ["", m.length + (n ? 1 : 0)] : [String.fromCodePoint(w), m.length + (n ? 1 : 0)];
    }
  }
  var d = /\\/;
  function g(y) {
    var p = d.test(y);
    if (!p)
      return y;
    for (var m = "", n = 0; n < y.length; n++) {
      if (y[n] === "\\") {
        var o = b(y.slice(n + 1, n + 7));
        if (o !== void 0) {
          m += o[0], n += o[1];
          continue;
        }
        if (y[n + 1] === "\\") {
          m += "\\", n++;
          continue;
        }
        y.length === n + 1 && (m += y[n]);
        continue;
      }
      m += y[n];
    }
    return m;
  }
  a.exports = t.default;
})(Ot, Ot.exports);
var wr = Ot.exports, Et = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = b;
  function b(d) {
    for (var g = arguments.length, y = new Array(g > 1 ? g - 1 : 0), p = 1; p < g; p++)
      y[p - 1] = arguments[p];
    for (; y.length > 0; ) {
      var m = y.shift();
      if (!d[m])
        return;
      d = d[m];
    }
    return d;
  }
  a.exports = t.default;
})(Et, Et.exports);
var fi = Et.exports, kt = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = b;
  function b(d) {
    for (var g = arguments.length, y = new Array(g > 1 ? g - 1 : 0), p = 1; p < g; p++)
      y[p - 1] = arguments[p];
    for (; y.length > 0; ) {
      var m = y.shift();
      d[m] || (d[m] = {}), d = d[m];
    }
  }
  a.exports = t.default;
})(kt, kt.exports);
var ci = kt.exports, xt = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = b;
  function b(d) {
    for (var g = "", y = d.indexOf("/*"), p = 0; y >= 0; ) {
      g = g + d.slice(p, y);
      var m = d.indexOf("*/", y + 2);
      if (m < 0)
        return g;
      p = m + 2, y = d.indexOf("/*", p);
    }
    return g = g + d.slice(p), g;
  }
  a.exports = t.default;
})(xt, xt.exports);
var pi = xt.exports;
ye.__esModule = !0;
ye.stripComments = ye.ensureObject = ye.getProp = ye.unesc = void 0;
var di = tt(wr);
ye.unesc = di.default;
var hi = tt(fi);
ye.getProp = hi.default;
var gi = tt(ci);
ye.ensureObject = gi.default;
var vi = tt(pi);
ye.stripComments = vi.default;
function tt(a) {
  return a && a.__esModule ? a : { default: a };
}
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = ye;
  function d(m, n) {
    for (var o = 0; o < n.length; o++) {
      var l = n[o];
      l.enumerable = l.enumerable || !1, l.configurable = !0, "value" in l && (l.writable = !0), Object.defineProperty(m, l.key, l);
    }
  }
  function g(m, n, o) {
    return n && d(m.prototype, n), o && d(m, o), m;
  }
  var y = function m(n, o) {
    if (typeof n != "object" || n === null)
      return n;
    var l = new n.constructor();
    for (var f in n)
      if (n.hasOwnProperty(f)) {
        var w = n[f], A = typeof w;
        f === "parent" && A === "object" ? o && (l[f] = o) : w instanceof Array ? l[f] = w.map(function(R) {
          return m(R, l);
        }) : l[f] = m(w, l);
      }
    return l;
  }, p = /* @__PURE__ */ function() {
    function m(o) {
      o === void 0 && (o = {}), Object.assign(this, o), this.spaces = this.spaces || {}, this.spaces.before = this.spaces.before || "", this.spaces.after = this.spaces.after || "";
    }
    var n = m.prototype;
    return n.remove = function() {
      return this.parent && this.parent.removeChild(this), this.parent = void 0, this;
    }, n.replaceWith = function() {
      if (this.parent) {
        for (var l in arguments)
          this.parent.insertBefore(this, arguments[l]);
        this.remove();
      }
      return this;
    }, n.next = function() {
      return this.parent.at(this.parent.index(this) + 1);
    }, n.prev = function() {
      return this.parent.at(this.parent.index(this) - 1);
    }, n.clone = function(l) {
      l === void 0 && (l = {});
      var f = y(this);
      for (var w in l)
        f[w] = l[w];
      return f;
    }, n.appendToPropertyAndEscape = function(l, f, w) {
      this.raws || (this.raws = {});
      var A = this[l], R = this.raws[l];
      this[l] = A + f, R || w !== f ? this.raws[l] = (R || A) + w : delete this.raws[l];
    }, n.setPropertyAndEscape = function(l, f, w) {
      this.raws || (this.raws = {}), this[l] = f, this.raws[l] = w;
    }, n.setPropertyWithoutEscape = function(l, f) {
      this[l] = f, this.raws && delete this.raws[l];
    }, n.isAtPosition = function(l, f) {
      if (this.source && this.source.start && this.source.end)
        return !(this.source.start.line > l || this.source.end.line < l || this.source.start.line === l && this.source.start.column > f || this.source.end.line === l && this.source.end.column < f);
    }, n.stringifyProperty = function(l) {
      return this.raws && this.raws[l] || this[l];
    }, n.valueToString = function() {
      return String(this.stringifyProperty("value"));
    }, n.toString = function() {
      return [this.rawSpaceBefore, this.valueToString(), this.rawSpaceAfter].join("");
    }, g(m, [{
      key: "rawSpaceBefore",
      get: function() {
        var l = this.raws && this.raws.spaces && this.raws.spaces.before;
        return l === void 0 && (l = this.spaces && this.spaces.before), l || "";
      },
      set: function(l) {
        (0, b.ensureObject)(this, "raws", "spaces"), this.raws.spaces.before = l;
      }
    }, {
      key: "rawSpaceAfter",
      get: function() {
        var l = this.raws && this.raws.spaces && this.raws.spaces.after;
        return l === void 0 && (l = this.spaces.after), l || "";
      },
      set: function(l) {
        (0, b.ensureObject)(this, "raws", "spaces"), this.raws.spaces.after = l;
      }
    }]), m;
  }();
  t.default = p, a.exports = t.default;
})(St, St.exports);
var De = St.exports, G = {};
G.__esModule = !0;
G.UNIVERSAL = G.ATTRIBUTE = G.CLASS = G.COMBINATOR = G.COMMENT = G.ID = G.NESTING = G.PSEUDO = G.ROOT = G.SELECTOR = G.STRING = G.TAG = void 0;
var mi = "tag";
G.TAG = mi;
var wi = "string";
G.STRING = wi;
var bi = "selector";
G.SELECTOR = bi;
var _i = "root";
G.ROOT = _i;
var Ti = "pseudo";
G.PSEUDO = Ti;
var yi = "nesting";
G.NESTING = yi;
var Si = "id";
G.ID = Si;
var Oi = "comment";
G.COMMENT = Oi;
var Ei = "combinator";
G.COMBINATOR = Ei;
var ki = "class";
G.CLASS = ki;
var xi = "attribute";
G.ATTRIBUTE = xi;
var Pi = "universal";
G.UNIVERSAL = Pi;
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = p(De), d = y(G);
  function g() {
    if (typeof WeakMap != "function")
      return null;
    var i = /* @__PURE__ */ new WeakMap();
    return g = function() {
      return i;
    }, i;
  }
  function y(i) {
    if (i && i.__esModule)
      return i;
    if (i === null || typeof i != "object" && typeof i != "function")
      return { default: i };
    var h = g();
    if (h && h.has(i))
      return h.get(i);
    var S = {}, E = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var O in i)
      if (Object.prototype.hasOwnProperty.call(i, O)) {
        var x = E ? Object.getOwnPropertyDescriptor(i, O) : null;
        x && (x.get || x.set) ? Object.defineProperty(S, O, x) : S[O] = i[O];
      }
    return S.default = i, h && h.set(i, S), S;
  }
  function p(i) {
    return i && i.__esModule ? i : { default: i };
  }
  function m(i, h) {
    var S;
    if (typeof Symbol > "u" || i[Symbol.iterator] == null) {
      if (Array.isArray(i) || (S = n(i)) || h && i && typeof i.length == "number") {
        S && (i = S);
        var E = 0;
        return function() {
          return E >= i.length ? { done: !0 } : { done: !1, value: i[E++] };
        };
      }
      throw new TypeError(`Invalid attempt to iterate non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.`);
    }
    return S = i[Symbol.iterator](), S.next.bind(S);
  }
  function n(i, h) {
    if (i) {
      if (typeof i == "string")
        return o(i, h);
      var S = Object.prototype.toString.call(i).slice(8, -1);
      if (S === "Object" && i.constructor && (S = i.constructor.name), S === "Map" || S === "Set")
        return Array.from(i);
      if (S === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(S))
        return o(i, h);
    }
  }
  function o(i, h) {
    (h == null || h > i.length) && (h = i.length);
    for (var S = 0, E = new Array(h); S < h; S++)
      E[S] = i[S];
    return E;
  }
  function l(i, h) {
    for (var S = 0; S < h.length; S++) {
      var E = h[S];
      E.enumerable = E.enumerable || !1, E.configurable = !0, "value" in E && (E.writable = !0), Object.defineProperty(i, E.key, E);
    }
  }
  function f(i, h, S) {
    return h && l(i.prototype, h), S && l(i, S), i;
  }
  function w(i, h) {
    i.prototype = Object.create(h.prototype), i.prototype.constructor = i, A(i, h);
  }
  function A(i, h) {
    return A = Object.setPrototypeOf || function(E, O) {
      return E.__proto__ = O, E;
    }, A(i, h);
  }
  var R = /* @__PURE__ */ function(i) {
    w(h, i);
    function h(E) {
      var O;
      return O = i.call(this, E) || this, O.nodes || (O.nodes = []), O;
    }
    var S = h.prototype;
    return S.append = function(O) {
      return O.parent = this, this.nodes.push(O), this;
    }, S.prepend = function(O) {
      return O.parent = this, this.nodes.unshift(O), this;
    }, S.at = function(O) {
      return this.nodes[O];
    }, S.index = function(O) {
      return typeof O == "number" ? O : this.nodes.indexOf(O);
    }, S.removeChild = function(O) {
      O = this.index(O), this.at(O).parent = void 0, this.nodes.splice(O, 1);
      var x;
      for (var T in this.indexes)
        x = this.indexes[T], x >= O && (this.indexes[T] = x - 1);
      return this;
    }, S.removeAll = function() {
      for (var O = m(this.nodes), x; !(x = O()).done; ) {
        var T = x.value;
        T.parent = void 0;
      }
      return this.nodes = [], this;
    }, S.empty = function() {
      return this.removeAll();
    }, S.insertAfter = function(O, x) {
      x.parent = this;
      var T = this.index(O);
      this.nodes.splice(T + 1, 0, x), x.parent = this;
      var P;
      for (var C in this.indexes)
        P = this.indexes[C], T <= P && (this.indexes[C] = P + 1);
      return this;
    }, S.insertBefore = function(O, x) {
      x.parent = this;
      var T = this.index(O);
      this.nodes.splice(T, 0, x), x.parent = this;
      var P;
      for (var C in this.indexes)
        P = this.indexes[C], P <= T && (this.indexes[C] = P + 1);
      return this;
    }, S._findChildAtPosition = function(O, x) {
      var T = void 0;
      return this.each(function(P) {
        if (P.atPosition) {
          var C = P.atPosition(O, x);
          if (C)
            return T = C, !1;
        } else if (P.isAtPosition(O, x))
          return T = P, !1;
      }), T;
    }, S.atPosition = function(O, x) {
      if (this.isAtPosition(O, x))
        return this._findChildAtPosition(O, x) || this;
    }, S._inferEndPosition = function() {
      this.last && this.last.source && this.last.source.end && (this.source = this.source || {}, this.source.end = this.source.end || {}, Object.assign(this.source.end, this.last.source.end));
    }, S.each = function(O) {
      this.lastEach || (this.lastEach = 0), this.indexes || (this.indexes = {}), this.lastEach++;
      var x = this.lastEach;
      if (this.indexes[x] = 0, !!this.length) {
        for (var T, P; this.indexes[x] < this.length && (T = this.indexes[x], P = O(this.at(T), T), P !== !1); )
          this.indexes[x] += 1;
        if (delete this.indexes[x], P === !1)
          return !1;
      }
    }, S.walk = function(O) {
      return this.each(function(x, T) {
        var P = O(x, T);
        if (P !== !1 && x.length && (P = x.walk(O)), P === !1)
          return !1;
      });
    }, S.walkAttributes = function(O) {
      var x = this;
      return this.walk(function(T) {
        if (T.type === d.ATTRIBUTE)
          return O.call(x, T);
      });
    }, S.walkClasses = function(O) {
      var x = this;
      return this.walk(function(T) {
        if (T.type === d.CLASS)
          return O.call(x, T);
      });
    }, S.walkCombinators = function(O) {
      var x = this;
      return this.walk(function(T) {
        if (T.type === d.COMBINATOR)
          return O.call(x, T);
      });
    }, S.walkComments = function(O) {
      var x = this;
      return this.walk(function(T) {
        if (T.type === d.COMMENT)
          return O.call(x, T);
      });
    }, S.walkIds = function(O) {
      var x = this;
      return this.walk(function(T) {
        if (T.type === d.ID)
          return O.call(x, T);
      });
    }, S.walkNesting = function(O) {
      var x = this;
      return this.walk(function(T) {
        if (T.type === d.NESTING)
          return O.call(x, T);
      });
    }, S.walkPseudos = function(O) {
      var x = this;
      return this.walk(function(T) {
        if (T.type === d.PSEUDO)
          return O.call(x, T);
      });
    }, S.walkTags = function(O) {
      var x = this;
      return this.walk(function(T) {
        if (T.type === d.TAG)
          return O.call(x, T);
      });
    }, S.walkUniversals = function(O) {
      var x = this;
      return this.walk(function(T) {
        if (T.type === d.UNIVERSAL)
          return O.call(x, T);
      });
    }, S.split = function(O) {
      var x = this, T = [];
      return this.reduce(function(P, C, B) {
        var I = O.call(x, C);
        return T.push(C), I ? (P.push(T), T = []) : B === x.length - 1 && P.push(T), P;
      }, []);
    }, S.map = function(O) {
      return this.nodes.map(O);
    }, S.reduce = function(O, x) {
      return this.nodes.reduce(O, x);
    }, S.every = function(O) {
      return this.nodes.every(O);
    }, S.some = function(O) {
      return this.nodes.some(O);
    }, S.filter = function(O) {
      return this.nodes.filter(O);
    }, S.sort = function(O) {
      return this.nodes.sort(O);
    }, S.toString = function() {
      return this.map(String).join("");
    }, f(h, [{
      key: "first",
      get: function() {
        return this.at(0);
      }
    }, {
      key: "last",
      get: function() {
        return this.at(this.length - 1);
      }
    }, {
      key: "length",
      get: function() {
        return this.nodes.length;
      }
    }]), h;
  }(b.default);
  t.default = R, a.exports = t.default;
})(yt, yt.exports);
var Ut = yt.exports;
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = g(Ut), d = G;
  function g(l) {
    return l && l.__esModule ? l : { default: l };
  }
  function y(l, f) {
    for (var w = 0; w < f.length; w++) {
      var A = f[w];
      A.enumerable = A.enumerable || !1, A.configurable = !0, "value" in A && (A.writable = !0), Object.defineProperty(l, A.key, A);
    }
  }
  function p(l, f, w) {
    return f && y(l.prototype, f), w && y(l, w), l;
  }
  function m(l, f) {
    l.prototype = Object.create(f.prototype), l.prototype.constructor = l, n(l, f);
  }
  function n(l, f) {
    return n = Object.setPrototypeOf || function(A, R) {
      return A.__proto__ = R, A;
    }, n(l, f);
  }
  var o = /* @__PURE__ */ function(l) {
    m(f, l);
    function f(A) {
      var R;
      return R = l.call(this, A) || this, R.type = d.ROOT, R;
    }
    var w = f.prototype;
    return w.toString = function() {
      var R = this.reduce(function(i, h) {
        return i.push(String(h)), i;
      }, []).join(",");
      return this.trailingComma ? R + "," : R;
    }, w.error = function(R, i) {
      return this._error ? this._error(R, i) : new Error(R);
    }, p(f, [{
      key: "errorGenerator",
      set: function(R) {
        this._error = R;
      }
    }]), f;
  }(b.default);
  t.default = o, a.exports = t.default;
})(Tt, Tt.exports);
var br = Tt.exports, Pt = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = g(Ut), d = G;
  function g(n) {
    return n && n.__esModule ? n : { default: n };
  }
  function y(n, o) {
    n.prototype = Object.create(o.prototype), n.prototype.constructor = n, p(n, o);
  }
  function p(n, o) {
    return p = Object.setPrototypeOf || function(f, w) {
      return f.__proto__ = w, f;
    }, p(n, o);
  }
  var m = /* @__PURE__ */ function(n) {
    y(o, n);
    function o(l) {
      var f;
      return f = n.call(this, l) || this, f.type = d.SELECTOR, f;
    }
    return o;
  }(b.default);
  t.default = m, a.exports = t.default;
})(Pt, Pt.exports);
var _r = Pt.exports, Lt = { exports: {} };
/*! https://mths.be/cssesc v3.0.0 by @mathias */
var Li = {}, Ii = Li.hasOwnProperty, Ai = function(t, b) {
  if (!t)
    return b;
  var d = {};
  for (var g in b)
    d[g] = Ii.call(t, g) ? t[g] : b[g];
  return d;
}, Bi = /[ -,\.\/:-@\[-\^`\{-~]/, Di = /[ -,\.\/:-@\[\]\^`\{-~]/, Ri = /(^|\\+)?(\\[A-F0-9]{1,6})\x20(?![a-fA-F0-9\x20])/g, $t = function a(t, b) {
  b = Ai(b, a.options), b.quotes != "single" && b.quotes != "double" && (b.quotes = "single");
  for (var d = b.quotes == "double" ? '"' : "'", g = b.isIdentifier, y = t.charAt(0), p = "", m = 0, n = t.length; m < n; ) {
    var o = t.charAt(m++), l = o.charCodeAt(), f = void 0;
    if (l < 32 || l > 126) {
      if (l >= 55296 && l <= 56319 && m < n) {
        var w = t.charCodeAt(m++);
        (w & 64512) == 56320 ? l = ((l & 1023) << 10) + (w & 1023) + 65536 : m--;
      }
      f = "\\" + l.toString(16).toUpperCase() + " ";
    } else
      b.escapeEverything ? Bi.test(o) ? f = "\\" + o : f = "\\" + l.toString(16).toUpperCase() + " " : /[\t\n\f\r\x0B]/.test(o) ? f = "\\" + l.toString(16).toUpperCase() + " " : o == "\\" || !g && (o == '"' && d == o || o == "'" && d == o) || g && Di.test(o) ? f = "\\" + o : f = o;
    p += f;
  }
  return g && (/^-[-\d]/.test(p) ? p = "\\-" + p.slice(1) : /\d/.test(y) && (p = "\\3" + y + " " + p.slice(1))), p = p.replace(Ri, function(A, R, i) {
    return R && R.length % 2 ? A : (R || "") + i;
  }), !g && b.wrap ? d + p + d : p;
};
$t.options = {
  escapeEverything: !1,
  isIdentifier: !1,
  quotes: "single",
  wrap: !1
};
$t.version = "3.0.0";
var Ht = $t;
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = p(Ht), d = ye, g = p(De), y = G;
  function p(w) {
    return w && w.__esModule ? w : { default: w };
  }
  function m(w, A) {
    for (var R = 0; R < A.length; R++) {
      var i = A[R];
      i.enumerable = i.enumerable || !1, i.configurable = !0, "value" in i && (i.writable = !0), Object.defineProperty(w, i.key, i);
    }
  }
  function n(w, A, R) {
    return A && m(w.prototype, A), R && m(w, R), w;
  }
  function o(w, A) {
    w.prototype = Object.create(A.prototype), w.prototype.constructor = w, l(w, A);
  }
  function l(w, A) {
    return l = Object.setPrototypeOf || function(i, h) {
      return i.__proto__ = h, i;
    }, l(w, A);
  }
  var f = /* @__PURE__ */ function(w) {
    o(A, w);
    function A(i) {
      var h;
      return h = w.call(this, i) || this, h.type = y.CLASS, h._constructed = !0, h;
    }
    var R = A.prototype;
    return R.valueToString = function() {
      return "." + w.prototype.valueToString.call(this);
    }, n(A, [{
      key: "value",
      get: function() {
        return this._value;
      },
      set: function(h) {
        if (this._constructed) {
          var S = (0, b.default)(h, {
            isIdentifier: !0
          });
          S !== h ? ((0, d.ensureObject)(this, "raws"), this.raws.value = S) : this.raws && delete this.raws.value;
        }
        this._value = h;
      }
    }]), A;
  }(g.default);
  t.default = f, a.exports = t.default;
})(Lt, Lt.exports);
var Tr = Lt.exports, It = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = g(De), d = G;
  function g(n) {
    return n && n.__esModule ? n : { default: n };
  }
  function y(n, o) {
    n.prototype = Object.create(o.prototype), n.prototype.constructor = n, p(n, o);
  }
  function p(n, o) {
    return p = Object.setPrototypeOf || function(f, w) {
      return f.__proto__ = w, f;
    }, p(n, o);
  }
  var m = /* @__PURE__ */ function(n) {
    y(o, n);
    function o(l) {
      var f;
      return f = n.call(this, l) || this, f.type = d.COMMENT, f;
    }
    return o;
  }(b.default);
  t.default = m, a.exports = t.default;
})(It, It.exports);
var yr = It.exports, At = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = g(De), d = G;
  function g(n) {
    return n && n.__esModule ? n : { default: n };
  }
  function y(n, o) {
    n.prototype = Object.create(o.prototype), n.prototype.constructor = n, p(n, o);
  }
  function p(n, o) {
    return p = Object.setPrototypeOf || function(f, w) {
      return f.__proto__ = w, f;
    }, p(n, o);
  }
  var m = /* @__PURE__ */ function(n) {
    y(o, n);
    function o(f) {
      var w;
      return w = n.call(this, f) || this, w.type = d.ID, w;
    }
    var l = o.prototype;
    return l.valueToString = function() {
      return "#" + n.prototype.valueToString.call(this);
    }, o;
  }(b.default);
  t.default = m, a.exports = t.default;
})(At, At.exports);
var Sr = At.exports, Bt = { exports: {} }, Dt = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = y(Ht), d = ye, g = y(De);
  function y(f) {
    return f && f.__esModule ? f : { default: f };
  }
  function p(f, w) {
    for (var A = 0; A < w.length; A++) {
      var R = w[A];
      R.enumerable = R.enumerable || !1, R.configurable = !0, "value" in R && (R.writable = !0), Object.defineProperty(f, R.key, R);
    }
  }
  function m(f, w, A) {
    return w && p(f.prototype, w), A && p(f, A), f;
  }
  function n(f, w) {
    f.prototype = Object.create(w.prototype), f.prototype.constructor = f, o(f, w);
  }
  function o(f, w) {
    return o = Object.setPrototypeOf || function(R, i) {
      return R.__proto__ = i, R;
    }, o(f, w);
  }
  var l = /* @__PURE__ */ function(f) {
    n(w, f);
    function w() {
      return f.apply(this, arguments) || this;
    }
    var A = w.prototype;
    return A.qualifiedName = function(i) {
      return this.namespace ? this.namespaceString + "|" + i : i;
    }, A.valueToString = function() {
      return this.qualifiedName(f.prototype.valueToString.call(this));
    }, m(w, [{
      key: "namespace",
      get: function() {
        return this._namespace;
      },
      set: function(i) {
        if (i === !0 || i === "*" || i === "&") {
          this._namespace = i, this.raws && delete this.raws.namespace;
          return;
        }
        var h = (0, b.default)(i, {
          isIdentifier: !0
        });
        this._namespace = i, h !== i ? ((0, d.ensureObject)(this, "raws"), this.raws.namespace = h) : this.raws && delete this.raws.namespace;
      }
    }, {
      key: "ns",
      get: function() {
        return this._namespace;
      },
      set: function(i) {
        this.namespace = i;
      }
    }, {
      key: "namespaceString",
      get: function() {
        if (this.namespace) {
          var i = this.stringifyProperty("namespace");
          return i === !0 ? "" : i;
        } else
          return "";
      }
    }]), w;
  }(g.default);
  t.default = l, a.exports = t.default;
})(Dt, Dt.exports);
var Gt = Dt.exports;
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = g(Gt), d = G;
  function g(n) {
    return n && n.__esModule ? n : { default: n };
  }
  function y(n, o) {
    n.prototype = Object.create(o.prototype), n.prototype.constructor = n, p(n, o);
  }
  function p(n, o) {
    return p = Object.setPrototypeOf || function(f, w) {
      return f.__proto__ = w, f;
    }, p(n, o);
  }
  var m = /* @__PURE__ */ function(n) {
    y(o, n);
    function o(l) {
      var f;
      return f = n.call(this, l) || this, f.type = d.TAG, f;
    }
    return o;
  }(b.default);
  t.default = m, a.exports = t.default;
})(Bt, Bt.exports);
var Or = Bt.exports, Rt = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = g(De), d = G;
  function g(n) {
    return n && n.__esModule ? n : { default: n };
  }
  function y(n, o) {
    n.prototype = Object.create(o.prototype), n.prototype.constructor = n, p(n, o);
  }
  function p(n, o) {
    return p = Object.setPrototypeOf || function(f, w) {
      return f.__proto__ = w, f;
    }, p(n, o);
  }
  var m = /* @__PURE__ */ function(n) {
    y(o, n);
    function o(l) {
      var f;
      return f = n.call(this, l) || this, f.type = d.STRING, f;
    }
    return o;
  }(b.default);
  t.default = m, a.exports = t.default;
})(Rt, Rt.exports);
var Er = Rt.exports, Ft = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = g(Ut), d = G;
  function g(n) {
    return n && n.__esModule ? n : { default: n };
  }
  function y(n, o) {
    n.prototype = Object.create(o.prototype), n.prototype.constructor = n, p(n, o);
  }
  function p(n, o) {
    return p = Object.setPrototypeOf || function(f, w) {
      return f.__proto__ = w, f;
    }, p(n, o);
  }
  var m = /* @__PURE__ */ function(n) {
    y(o, n);
    function o(f) {
      var w;
      return w = n.call(this, f) || this, w.type = d.PSEUDO, w;
    }
    var l = o.prototype;
    return l.toString = function() {
      var w = this.length ? "(" + this.map(String).join(",") + ")" : "";
      return [this.rawSpaceBefore, this.stringifyProperty("value"), w, this.rawSpaceAfter].join("");
    }, o;
  }(b.default);
  t.default = m, a.exports = t.default;
})(Ft, Ft.exports);
var kr = Ft.exports, jt = {}, Fi = qi;
function qi(a, t) {
  if (mt("noDeprecation"))
    return a;
  var b = !1;
  function d() {
    if (!b) {
      if (mt("throwDeprecation"))
        throw new Error(t);
      mt("traceDeprecation") ? console.trace(t) : console.warn(t), b = !0;
    }
    return a.apply(this, arguments);
  }
  return d;
}
function mt(a) {
  try {
    if (!Ue.localStorage)
      return !1;
  } catch {
    return !1;
  }
  var t = Ue.localStorage[a];
  return t == null ? !1 : String(t).toLowerCase() === "true";
}
(function(a) {
  a.__esModule = !0, a.unescapeValue = h, a.default = void 0;
  var t = p(Ht), b = p(wr), d = p(Gt), g = G, y;
  function p(T) {
    return T && T.__esModule ? T : { default: T };
  }
  function m(T, P) {
    for (var C = 0; C < P.length; C++) {
      var B = P[C];
      B.enumerable = B.enumerable || !1, B.configurable = !0, "value" in B && (B.writable = !0), Object.defineProperty(T, B.key, B);
    }
  }
  function n(T, P, C) {
    return P && m(T.prototype, P), C && m(T, C), T;
  }
  function o(T, P) {
    T.prototype = Object.create(P.prototype), T.prototype.constructor = T, l(T, P);
  }
  function l(T, P) {
    return l = Object.setPrototypeOf || function(B, I) {
      return B.__proto__ = I, B;
    }, l(T, P);
  }
  var f = Fi, w = /^('|")([^]*)\1$/, A = f(function() {
  }, "Assigning an attribute a value containing characters that might need to be escaped is deprecated. Call attribute.setValue() instead."), R = f(function() {
  }, "Assigning attr.quoted is deprecated and has no effect. Assign to attr.quoteMark instead."), i = f(function() {
  }, "Constructing an Attribute selector with a value without specifying quoteMark is deprecated. Note: The value should be unescaped now.");
  function h(T) {
    var P = !1, C = null, B = T, I = B.match(w);
    return I && (C = I[1], B = I[2]), B = (0, b.default)(B), B !== T && (P = !0), {
      deprecatedUsage: P,
      unescaped: B,
      quoteMark: C
    };
  }
  function S(T) {
    if (T.quoteMark !== void 0 || T.value === void 0)
      return T;
    i();
    var P = h(T.value), C = P.quoteMark, B = P.unescaped;
    return T.raws || (T.raws = {}), T.raws.value === void 0 && (T.raws.value = T.value), T.value = B, T.quoteMark = C, T;
  }
  var E = /* @__PURE__ */ function(T) {
    o(P, T);
    function P(B) {
      var I;
      return B === void 0 && (B = {}), I = T.call(this, S(B)) || this, I.type = g.ATTRIBUTE, I.raws = I.raws || {}, Object.defineProperty(I.raws, "unquoted", {
        get: f(function() {
          return I.value;
        }, "attr.raws.unquoted is deprecated. Call attr.value instead."),
        set: f(function() {
          return I.value;
        }, "Setting attr.raws.unquoted is deprecated and has no effect. attr.value is unescaped by default now.")
      }), I._constructed = !0, I;
    }
    var C = P.prototype;
    return C.getQuotedValue = function(I) {
      I === void 0 && (I = {});
      var D = this._determineQuoteMark(I), Y = O[D], Z = (0, t.default)(this._value, Y);
      return Z;
    }, C._determineQuoteMark = function(I) {
      return I.smart ? this.smartQuoteMark(I) : this.preferredQuoteMark(I);
    }, C.setValue = function(I, D) {
      D === void 0 && (D = {}), this._value = I, this._quoteMark = this._determineQuoteMark(D), this._syncRawValue();
    }, C.smartQuoteMark = function(I) {
      var D = this.value, Y = D.replace(/[^']/g, "").length, Z = D.replace(/[^"]/g, "").length;
      if (Y + Z === 0) {
        var pe = (0, t.default)(D, {
          isIdentifier: !0
        });
        if (pe === D)
          return P.NO_QUOTE;
        var ue = this.preferredQuoteMark(I);
        if (ue === P.NO_QUOTE) {
          var fe = this.quoteMark || I.quoteMark || P.DOUBLE_QUOTE, j = O[fe], he = (0, t.default)(D, j);
          if (he.length < pe.length)
            return fe;
        }
        return ue;
      } else
        return Z === Y ? this.preferredQuoteMark(I) : Z < Y ? P.DOUBLE_QUOTE : P.SINGLE_QUOTE;
    }, C.preferredQuoteMark = function(I) {
      var D = I.preferCurrentQuoteMark ? this.quoteMark : I.quoteMark;
      return D === void 0 && (D = I.preferCurrentQuoteMark ? I.quoteMark : this.quoteMark), D === void 0 && (D = P.DOUBLE_QUOTE), D;
    }, C._syncRawValue = function() {
      var I = (0, t.default)(this._value, O[this.quoteMark]);
      I === this._value ? this.raws && delete this.raws.value : this.raws.value = I;
    }, C._handleEscapes = function(I, D) {
      if (this._constructed) {
        var Y = (0, t.default)(D, {
          isIdentifier: !0
        });
        Y !== D ? this.raws[I] = Y : delete this.raws[I];
      }
    }, C._spacesFor = function(I) {
      var D = {
        before: "",
        after: ""
      }, Y = this.spaces[I] || {}, Z = this.raws.spaces && this.raws.spaces[I] || {};
      return Object.assign(D, Y, Z);
    }, C._stringFor = function(I, D, Y) {
      D === void 0 && (D = I), Y === void 0 && (Y = x);
      var Z = this._spacesFor(D);
      return Y(this.stringifyProperty(I), Z);
    }, C.offsetOf = function(I) {
      var D = 1, Y = this._spacesFor("attribute");
      if (D += Y.before.length, I === "namespace" || I === "ns")
        return this.namespace ? D : -1;
      if (I === "attributeNS" || (D += this.namespaceString.length, this.namespace && (D += 1), I === "attribute"))
        return D;
      D += this.stringifyProperty("attribute").length, D += Y.after.length;
      var Z = this._spacesFor("operator");
      D += Z.before.length;
      var pe = this.stringifyProperty("operator");
      if (I === "operator")
        return pe ? D : -1;
      D += pe.length, D += Z.after.length;
      var ue = this._spacesFor("value");
      D += ue.before.length;
      var fe = this.stringifyProperty("value");
      if (I === "value")
        return fe ? D : -1;
      D += fe.length, D += ue.after.length;
      var j = this._spacesFor("insensitive");
      return D += j.before.length, I === "insensitive" && this.insensitive ? D : -1;
    }, C.toString = function() {
      var I = this, D = [this.rawSpaceBefore, "["];
      return D.push(this._stringFor("qualifiedAttribute", "attribute")), this.operator && (this.value || this.value === "") && (D.push(this._stringFor("operator")), D.push(this._stringFor("value")), D.push(this._stringFor("insensitiveFlag", "insensitive", function(Y, Z) {
        return Y.length > 0 && !I.quoted && Z.before.length === 0 && !(I.spaces.value && I.spaces.value.after) && (Z.before = " "), x(Y, Z);
      }))), D.push("]"), D.push(this.rawSpaceAfter), D.join("");
    }, n(P, [{
      key: "quoted",
      get: function() {
        var I = this.quoteMark;
        return I === "'" || I === '"';
      },
      set: function(I) {
        R();
      }
      /**
       * returns a single (`'`) or double (`"`) quote character if the value is quoted.
       * returns `null` if the value is not quoted.
       * returns `undefined` if the quotation state is unknown (this can happen when
       * the attribute is constructed without specifying a quote mark.)
       */
    }, {
      key: "quoteMark",
      get: function() {
        return this._quoteMark;
      },
      set: function(I) {
        if (!this._constructed) {
          this._quoteMark = I;
          return;
        }
        this._quoteMark !== I && (this._quoteMark = I, this._syncRawValue());
      }
    }, {
      key: "qualifiedAttribute",
      get: function() {
        return this.qualifiedName(this.raws.attribute || this.attribute);
      }
    }, {
      key: "insensitiveFlag",
      get: function() {
        return this.insensitive ? "i" : "";
      }
    }, {
      key: "value",
      get: function() {
        return this._value;
      },
      set: function(I) {
        if (this._constructed) {
          var D = h(I), Y = D.deprecatedUsage, Z = D.unescaped, pe = D.quoteMark;
          if (Y && A(), Z === this._value && pe === this._quoteMark)
            return;
          this._value = Z, this._quoteMark = pe, this._syncRawValue();
        } else
          this._value = I;
      }
    }, {
      key: "attribute",
      get: function() {
        return this._attribute;
      },
      set: function(I) {
        this._handleEscapes("attribute", I), this._attribute = I;
      }
    }]), P;
  }(d.default);
  a.default = E, E.NO_QUOTE = null, E.SINGLE_QUOTE = "'", E.DOUBLE_QUOTE = '"';
  var O = (y = {
    "'": {
      quotes: "single",
      wrap: !0
    },
    '"': {
      quotes: "double",
      wrap: !0
    }
  }, y[null] = {
    isIdentifier: !0
  }, y);
  function x(T, P) {
    return "" + P.before + T + P.after;
  }
})(jt);
var qt = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = g(Gt), d = G;
  function g(n) {
    return n && n.__esModule ? n : { default: n };
  }
  function y(n, o) {
    n.prototype = Object.create(o.prototype), n.prototype.constructor = n, p(n, o);
  }
  function p(n, o) {
    return p = Object.setPrototypeOf || function(f, w) {
      return f.__proto__ = w, f;
    }, p(n, o);
  }
  var m = /* @__PURE__ */ function(n) {
    y(o, n);
    function o(l) {
      var f;
      return f = n.call(this, l) || this, f.type = d.UNIVERSAL, f.value = "*", f;
    }
    return o;
  }(b.default);
  t.default = m, a.exports = t.default;
})(qt, qt.exports);
var xr = qt.exports, Mt = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = g(De), d = G;
  function g(n) {
    return n && n.__esModule ? n : { default: n };
  }
  function y(n, o) {
    n.prototype = Object.create(o.prototype), n.prototype.constructor = n, p(n, o);
  }
  function p(n, o) {
    return p = Object.setPrototypeOf || function(f, w) {
      return f.__proto__ = w, f;
    }, p(n, o);
  }
  var m = /* @__PURE__ */ function(n) {
    y(o, n);
    function o(l) {
      var f;
      return f = n.call(this, l) || this, f.type = d.COMBINATOR, f;
    }
    return o;
  }(b.default);
  t.default = m, a.exports = t.default;
})(Mt, Mt.exports);
var Pr = Mt.exports, Nt = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = g(De), d = G;
  function g(n) {
    return n && n.__esModule ? n : { default: n };
  }
  function y(n, o) {
    n.prototype = Object.create(o.prototype), n.prototype.constructor = n, p(n, o);
  }
  function p(n, o) {
    return p = Object.setPrototypeOf || function(f, w) {
      return f.__proto__ = w, f;
    }, p(n, o);
  }
  var m = /* @__PURE__ */ function(n) {
    y(o, n);
    function o(l) {
      var f;
      return f = n.call(this, l) || this, f.type = d.NESTING, f.value = "&", f;
    }
    return o;
  }(b.default);
  t.default = m, a.exports = t.default;
})(Nt, Nt.exports);
var Lr = Nt.exports, Ct = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = b;
  function b(d) {
    return d.sort(function(g, y) {
      return g - y;
    });
  }
  a.exports = t.default;
})(Ct, Ct.exports);
var Mi = Ct.exports, Ir = {}, M = {};
M.__esModule = !0;
M.combinator = M.word = M.comment = M.str = M.tab = M.newline = M.feed = M.cr = M.backslash = M.bang = M.slash = M.doubleQuote = M.singleQuote = M.space = M.greaterThan = M.pipe = M.equals = M.plus = M.caret = M.tilde = M.dollar = M.closeSquare = M.openSquare = M.closeParenthesis = M.openParenthesis = M.semicolon = M.colon = M.comma = M.at = M.asterisk = M.ampersand = void 0;
var Ni = 38;
M.ampersand = Ni;
var Ci = 42;
M.asterisk = Ci;
var zi = 64;
M.at = zi;
var Wi = 44;
M.comma = Wi;
var Ui = 58;
M.colon = Ui;
var $i = 59;
M.semicolon = $i;
var Hi = 40;
M.openParenthesis = Hi;
var Gi = 41;
M.closeParenthesis = Gi;
var ji = 91;
M.openSquare = ji;
var Qi = 93;
M.closeSquare = Qi;
var Yi = 36;
M.dollar = Yi;
var Vi = 126;
M.tilde = Vi;
var Ki = 94;
M.caret = Ki;
var Ji = 43;
M.plus = Ji;
var Xi = 61;
M.equals = Xi;
var Zi = 124;
M.pipe = Zi;
var eo = 62;
M.greaterThan = eo;
var to = 32;
M.space = to;
var Ar = 39;
M.singleQuote = Ar;
var ro = 34;
M.doubleQuote = ro;
var no = 47;
M.slash = no;
var io = 33;
M.bang = io;
var oo = 92;
M.backslash = oo;
var ao = 13;
M.cr = ao;
var so = 12;
M.feed = so;
var uo = 10;
M.newline = uo;
var lo = 9;
M.tab = lo;
var fo = Ar;
M.str = fo;
var co = -1;
M.comment = co;
var po = -2;
M.word = po;
var ho = -3;
M.combinator = ho;
(function(a) {
  a.__esModule = !0, a.default = R, a.FIELDS = void 0;
  var t = y(M), b, d;
  function g() {
    if (typeof WeakMap != "function")
      return null;
    var i = /* @__PURE__ */ new WeakMap();
    return g = function() {
      return i;
    }, i;
  }
  function y(i) {
    if (i && i.__esModule)
      return i;
    if (i === null || typeof i != "object" && typeof i != "function")
      return { default: i };
    var h = g();
    if (h && h.has(i))
      return h.get(i);
    var S = {}, E = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var O in i)
      if (Object.prototype.hasOwnProperty.call(i, O)) {
        var x = E ? Object.getOwnPropertyDescriptor(i, O) : null;
        x && (x.get || x.set) ? Object.defineProperty(S, O, x) : S[O] = i[O];
      }
    return S.default = i, h && h.set(i, S), S;
  }
  for (var p = (b = {}, b[t.tab] = !0, b[t.newline] = !0, b[t.cr] = !0, b[t.feed] = !0, b), m = (d = {}, d[t.space] = !0, d[t.tab] = !0, d[t.newline] = !0, d[t.cr] = !0, d[t.feed] = !0, d[t.ampersand] = !0, d[t.asterisk] = !0, d[t.bang] = !0, d[t.comma] = !0, d[t.colon] = !0, d[t.semicolon] = !0, d[t.openParenthesis] = !0, d[t.closeParenthesis] = !0, d[t.openSquare] = !0, d[t.closeSquare] = !0, d[t.singleQuote] = !0, d[t.doubleQuote] = !0, d[t.plus] = !0, d[t.pipe] = !0, d[t.tilde] = !0, d[t.greaterThan] = !0, d[t.equals] = !0, d[t.dollar] = !0, d[t.caret] = !0, d[t.slash] = !0, d), n = {}, o = "0123456789abcdefABCDEF", l = 0; l < o.length; l++)
    n[o.charCodeAt(l)] = !0;
  function f(i, h) {
    var S = h, E;
    do {
      if (E = i.charCodeAt(S), m[E])
        return S - 1;
      E === t.backslash ? S = w(i, S) + 1 : S++;
    } while (S < i.length);
    return S - 1;
  }
  function w(i, h) {
    var S = h, E = i.charCodeAt(S + 1);
    if (!p[E])
      if (n[E]) {
        var O = 0;
        do
          S++, O++, E = i.charCodeAt(S + 1);
        while (n[E] && O < 6);
        O < 6 && E === t.space && S++;
      } else
        S++;
    return S;
  }
  var A = {
    TYPE: 0,
    START_LINE: 1,
    START_COL: 2,
    END_LINE: 3,
    END_COL: 4,
    START_POS: 5,
    END_POS: 6
  };
  a.FIELDS = A;
  function R(i) {
    var h = [], S = i.css.valueOf(), E = S, O = E.length, x = -1, T = 1, P = 0, C = 0, B, I, D, Y, Z, pe, ue, fe, j, he, Se, Re, Oe;
    function F(N, k) {
      if (i.safe)
        S += k, j = S.length - 1;
      else
        throw i.error("Unclosed " + N, T, P - x, P);
    }
    for (; P < O; ) {
      switch (B = S.charCodeAt(P), B === t.newline && (x = P, T += 1), B) {
        case t.space:
        case t.tab:
        case t.newline:
        case t.cr:
        case t.feed:
          j = P;
          do
            j += 1, B = S.charCodeAt(j), B === t.newline && (x = j, T += 1);
          while (B === t.space || B === t.newline || B === t.tab || B === t.cr || B === t.feed);
          Oe = t.space, Y = T, D = j - x - 1, C = j;
          break;
        case t.plus:
        case t.greaterThan:
        case t.tilde:
        case t.pipe:
          j = P;
          do
            j += 1, B = S.charCodeAt(j);
          while (B === t.plus || B === t.greaterThan || B === t.tilde || B === t.pipe);
          Oe = t.combinator, Y = T, D = P - x, C = j;
          break;
        case t.asterisk:
        case t.ampersand:
        case t.bang:
        case t.comma:
        case t.equals:
        case t.dollar:
        case t.caret:
        case t.openSquare:
        case t.closeSquare:
        case t.colon:
        case t.semicolon:
        case t.openParenthesis:
        case t.closeParenthesis:
          j = P, Oe = B, Y = T, D = P - x, C = j + 1;
          break;
        case t.singleQuote:
        case t.doubleQuote:
          Re = B === t.singleQuote ? "'" : '"', j = P;
          do
            for (Z = !1, j = S.indexOf(Re, j + 1), j === -1 && F("quote", Re), pe = j; S.charCodeAt(pe - 1) === t.backslash; )
              pe -= 1, Z = !Z;
          while (Z);
          Oe = t.str, Y = T, D = P - x, C = j + 1;
          break;
        default:
          B === t.slash && S.charCodeAt(P + 1) === t.asterisk ? (j = S.indexOf("*/", P + 2) + 1, j === 0 && F("comment", "*/"), I = S.slice(P, j + 1), fe = I.split(`
`), ue = fe.length - 1, ue > 0 ? (he = T + ue, Se = j - fe[ue].length) : (he = T, Se = x), Oe = t.comment, T = he, Y = he, D = j - Se) : B === t.slash ? (j = P, Oe = B, Y = T, D = P - x, C = j + 1) : (j = f(S, P), Oe = t.word, Y = T, D = j - x), C = j + 1;
          break;
      }
      h.push([
        Oe,
        // [0] Token type
        T,
        // [1] Starting line
        P - x,
        // [2] Starting column
        Y,
        // [3] Ending line
        D,
        // [4] Ending column
        P,
        // [5] Start position / Source index
        C
        // [6] End position
      ]), Se && (x = Se, Se = null), P = C;
    }
    return h;
  }
})(Ir);
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = C(br), d = C(_r), g = C(Tr), y = C(yr), p = C(Sr), m = C(Or), n = C(Er), o = C(kr), l = P(jt), f = C(xr), w = C(Pr), A = C(Lr), R = C(Mi), i = P(Ir), h = P(M), S = P(G), E = ye, O, x;
  function T() {
    if (typeof WeakMap != "function")
      return null;
    var F = /* @__PURE__ */ new WeakMap();
    return T = function() {
      return F;
    }, F;
  }
  function P(F) {
    if (F && F.__esModule)
      return F;
    if (F === null || typeof F != "object" && typeof F != "function")
      return { default: F };
    var N = T();
    if (N && N.has(F))
      return N.get(F);
    var k = {}, v = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var L in F)
      if (Object.prototype.hasOwnProperty.call(F, L)) {
        var z = v ? Object.getOwnPropertyDescriptor(F, L) : null;
        z && (z.get || z.set) ? Object.defineProperty(k, L, z) : k[L] = F[L];
      }
    return k.default = F, N && N.set(F, k), k;
  }
  function C(F) {
    return F && F.__esModule ? F : { default: F };
  }
  function B(F, N) {
    for (var k = 0; k < N.length; k++) {
      var v = N[k];
      v.enumerable = v.enumerable || !1, v.configurable = !0, "value" in v && (v.writable = !0), Object.defineProperty(F, v.key, v);
    }
  }
  function I(F, N, k) {
    return N && B(F.prototype, N), k && B(F, k), F;
  }
  var D = (O = {}, O[h.space] = !0, O[h.cr] = !0, O[h.feed] = !0, O[h.newline] = !0, O[h.tab] = !0, O), Y = Object.assign({}, D, (x = {}, x[h.comment] = !0, x));
  function Z(F) {
    return {
      line: F[i.FIELDS.START_LINE],
      column: F[i.FIELDS.START_COL]
    };
  }
  function pe(F) {
    return {
      line: F[i.FIELDS.END_LINE],
      column: F[i.FIELDS.END_COL]
    };
  }
  function ue(F, N, k, v) {
    return {
      start: {
        line: F,
        column: N
      },
      end: {
        line: k,
        column: v
      }
    };
  }
  function fe(F) {
    return ue(F[i.FIELDS.START_LINE], F[i.FIELDS.START_COL], F[i.FIELDS.END_LINE], F[i.FIELDS.END_COL]);
  }
  function j(F, N) {
    if (F)
      return ue(F[i.FIELDS.START_LINE], F[i.FIELDS.START_COL], N[i.FIELDS.END_LINE], N[i.FIELDS.END_COL]);
  }
  function he(F, N) {
    var k = F[N];
    if (typeof k == "string")
      return k.indexOf("\\") !== -1 && ((0, E.ensureObject)(F, "raws"), F[N] = (0, E.unesc)(k), F.raws[N] === void 0 && (F.raws[N] = k)), F;
  }
  function Se(F, N) {
    for (var k = -1, v = []; (k = F.indexOf(N, k + 1)) !== -1; )
      v.push(k);
    return v;
  }
  function Re() {
    var F = Array.prototype.concat.apply([], arguments);
    return F.filter(function(N, k) {
      return k === F.indexOf(N);
    });
  }
  var Oe = /* @__PURE__ */ function() {
    function F(k, v) {
      v === void 0 && (v = {}), this.rule = k, this.options = Object.assign({
        lossy: !1,
        safe: !1
      }, v), this.position = 0, this.css = typeof this.rule == "string" ? this.rule : this.rule.selector, this.tokens = (0, i.default)({
        css: this.css,
        error: this._errorGenerator(),
        safe: this.options.safe
      });
      var L = j(this.tokens[0], this.tokens[this.tokens.length - 1]);
      this.root = new b.default({
        source: L
      }), this.root.errorGenerator = this._errorGenerator();
      var z = new d.default({
        source: {
          start: {
            line: 1,
            column: 1
          }
        }
      });
      this.root.append(z), this.current = z, this.loop();
    }
    var N = F.prototype;
    return N._errorGenerator = function() {
      var v = this;
      return function(L, z) {
        return typeof v.rule == "string" ? new Error(L) : v.rule.error(L, z);
      };
    }, N.attribute = function() {
      var v = [], L = this.currToken;
      for (this.position++; this.position < this.tokens.length && this.currToken[i.FIELDS.TYPE] !== h.closeSquare; )
        v.push(this.currToken), this.position++;
      if (this.currToken[i.FIELDS.TYPE] !== h.closeSquare)
        return this.expected("closing square bracket", this.currToken[i.FIELDS.START_POS]);
      var z = v.length, _ = {
        source: ue(L[1], L[2], this.currToken[3], this.currToken[4]),
        sourceIndex: L[i.FIELDS.START_POS]
      };
      if (z === 1 && !~[h.word].indexOf(v[0][i.FIELDS.TYPE]))
        return this.expected("attribute", v[0][i.FIELDS.START_POS]);
      for (var $ = 0, J = "", Q = "", W = null, ee = !1; $ < z; ) {
        var le = v[$], H = this.content(le), ie = v[$ + 1];
        switch (le[i.FIELDS.TYPE]) {
          case h.space:
            if (ee = !0, this.options.lossy)
              break;
            if (W) {
              (0, E.ensureObject)(_, "spaces", W);
              var Le = _.spaces[W].after || "";
              _.spaces[W].after = Le + H;
              var Ee = (0, E.getProp)(_, "raws", "spaces", W, "after") || null;
              Ee && (_.raws.spaces[W].after = Ee + H);
            } else
              J = J + H, Q = Q + H;
            break;
          case h.asterisk:
            if (ie[i.FIELDS.TYPE] === h.equals)
              _.operator = H, W = "operator";
            else if ((!_.namespace || W === "namespace" && !ee) && ie) {
              J && ((0, E.ensureObject)(_, "spaces", "attribute"), _.spaces.attribute.before = J, J = ""), Q && ((0, E.ensureObject)(_, "raws", "spaces", "attribute"), _.raws.spaces.attribute.before = J, Q = ""), _.namespace = (_.namespace || "") + H;
              var ge = (0, E.getProp)(_, "raws", "namespace") || null;
              ge && (_.raws.namespace += H), W = "namespace";
            }
            ee = !1;
            break;
          case h.dollar:
            if (W === "value") {
              var ce = (0, E.getProp)(_, "raws", "value");
              _.value += "$", ce && (_.raws.value = ce + "$");
              break;
            }
          case h.caret:
            ie[i.FIELDS.TYPE] === h.equals && (_.operator = H, W = "operator"), ee = !1;
            break;
          case h.combinator:
            if (H === "~" && ie[i.FIELDS.TYPE] === h.equals && (_.operator = H, W = "operator"), H !== "|") {
              ee = !1;
              break;
            }
            ie[i.FIELDS.TYPE] === h.equals ? (_.operator = H, W = "operator") : !_.namespace && !_.attribute && (_.namespace = !0), ee = !1;
            break;
          case h.word:
            if (ie && this.content(ie) === "|" && v[$ + 2] && v[$ + 2][i.FIELDS.TYPE] !== h.equals && // this look-ahead probably fails with comment nodes involved.
            !_.operator && !_.namespace)
              _.namespace = H, W = "namespace";
            else if (!_.attribute || W === "attribute" && !ee) {
              J && ((0, E.ensureObject)(_, "spaces", "attribute"), _.spaces.attribute.before = J, J = ""), Q && ((0, E.ensureObject)(_, "raws", "spaces", "attribute"), _.raws.spaces.attribute.before = Q, Q = ""), _.attribute = (_.attribute || "") + H;
              var de = (0, E.getProp)(_, "raws", "attribute") || null;
              de && (_.raws.attribute += H), W = "attribute";
            } else if (!_.value && _.value !== "" || W === "value" && !ee) {
              var ve = (0, E.unesc)(H), _e = (0, E.getProp)(_, "raws", "value") || "", we = _.value || "";
              _.value = we + ve, _.quoteMark = null, (ve !== H || _e) && ((0, E.ensureObject)(_, "raws"), _.raws.value = (_e || we) + H), W = "value";
            } else {
              var Fe = H === "i" || H === "I";
              (_.value || _.value === "") && (_.quoteMark || ee) ? (_.insensitive = Fe, (!Fe || H === "I") && ((0, E.ensureObject)(_, "raws"), _.raws.insensitiveFlag = H), W = "insensitive", J && ((0, E.ensureObject)(_, "spaces", "insensitive"), _.spaces.insensitive.before = J, J = ""), Q && ((0, E.ensureObject)(_, "raws", "spaces", "insensitive"), _.raws.spaces.insensitive.before = Q, Q = "")) : (_.value || _.value === "") && (W = "value", _.value += H, _.raws.value && (_.raws.value += H));
            }
            ee = !1;
            break;
          case h.str:
            if (!_.attribute || !_.operator)
              return this.error("Expected an attribute followed by an operator preceding the string.", {
                index: le[i.FIELDS.START_POS]
              });
            var Be = (0, l.unescapeValue)(H), rt = Be.unescaped, nt = Be.quoteMark;
            _.value = rt, _.quoteMark = nt, W = "value", (0, E.ensureObject)(_, "raws"), _.raws.value = H, ee = !1;
            break;
          case h.equals:
            if (!_.attribute)
              return this.expected("attribute", le[i.FIELDS.START_POS], H);
            if (_.value)
              return this.error('Unexpected "=" found; an operator was already defined.', {
                index: le[i.FIELDS.START_POS]
              });
            _.operator = _.operator ? _.operator + H : H, W = "operator", ee = !1;
            break;
          case h.comment:
            if (W)
              if (ee || ie && ie[i.FIELDS.TYPE] === h.space || W === "insensitive") {
                var Ce = (0, E.getProp)(_, "spaces", W, "after") || "", je = (0, E.getProp)(_, "raws", "spaces", W, "after") || Ce;
                (0, E.ensureObject)(_, "raws", "spaces", W), _.raws.spaces[W].after = je + H;
              } else {
                var Qe = _[W] || "", Ye = (0, E.getProp)(_, "raws", W) || Qe;
                (0, E.ensureObject)(_, "raws"), _.raws[W] = Ye + H;
              }
            else
              Q = Q + H;
            break;
          default:
            return this.error('Unexpected "' + H + '" found.', {
              index: le[i.FIELDS.START_POS]
            });
        }
        $++;
      }
      he(_, "attribute"), he(_, "namespace"), this.newNode(new l.default(_)), this.position++;
    }, N.parseWhitespaceEquivalentTokens = function(v) {
      v < 0 && (v = this.tokens.length);
      var L = this.position, z = [], _ = "", $ = void 0;
      do
        if (D[this.currToken[i.FIELDS.TYPE]])
          this.options.lossy || (_ += this.content());
        else if (this.currToken[i.FIELDS.TYPE] === h.comment) {
          var J = {};
          _ && (J.before = _, _ = ""), $ = new y.default({
            value: this.content(),
            source: fe(this.currToken),
            sourceIndex: this.currToken[i.FIELDS.START_POS],
            spaces: J
          }), z.push($);
        }
      while (++this.position < v);
      if (_) {
        if ($)
          $.spaces.after = _;
        else if (!this.options.lossy) {
          var Q = this.tokens[L], W = this.tokens[this.position - 1];
          z.push(new n.default({
            value: "",
            source: ue(Q[i.FIELDS.START_LINE], Q[i.FIELDS.START_COL], W[i.FIELDS.END_LINE], W[i.FIELDS.END_COL]),
            sourceIndex: Q[i.FIELDS.START_POS],
            spaces: {
              before: _,
              after: ""
            }
          }));
        }
      }
      return z;
    }, N.convertWhitespaceNodesToSpace = function(v, L) {
      var z = this;
      L === void 0 && (L = !1);
      var _ = "", $ = "";
      v.forEach(function(Q) {
        var W = z.lossySpace(Q.spaces.before, L), ee = z.lossySpace(Q.rawSpaceBefore, L);
        _ += W + z.lossySpace(Q.spaces.after, L && W.length === 0), $ += W + Q.value + z.lossySpace(Q.rawSpaceAfter, L && ee.length === 0);
      }), $ === _ && ($ = void 0);
      var J = {
        space: _,
        rawSpace: $
      };
      return J;
    }, N.isNamedCombinator = function(v) {
      return v === void 0 && (v = this.position), this.tokens[v + 0] && this.tokens[v + 0][i.FIELDS.TYPE] === h.slash && this.tokens[v + 1] && this.tokens[v + 1][i.FIELDS.TYPE] === h.word && this.tokens[v + 2] && this.tokens[v + 2][i.FIELDS.TYPE] === h.slash;
    }, N.namedCombinator = function() {
      if (this.isNamedCombinator()) {
        var v = this.content(this.tokens[this.position + 1]), L = (0, E.unesc)(v).toLowerCase(), z = {};
        L !== v && (z.value = "/" + v + "/");
        var _ = new w.default({
          value: "/" + L + "/",
          source: ue(this.currToken[i.FIELDS.START_LINE], this.currToken[i.FIELDS.START_COL], this.tokens[this.position + 2][i.FIELDS.END_LINE], this.tokens[this.position + 2][i.FIELDS.END_COL]),
          sourceIndex: this.currToken[i.FIELDS.START_POS],
          raws: z
        });
        return this.position = this.position + 3, _;
      } else
        this.unexpected();
    }, N.combinator = function() {
      var v = this;
      if (this.content() === "|")
        return this.namespace();
      var L = this.locateNextMeaningfulToken(this.position);
      if (L < 0 || this.tokens[L][i.FIELDS.TYPE] === h.comma) {
        var z = this.parseWhitespaceEquivalentTokens(L);
        if (z.length > 0) {
          var _ = this.current.last;
          if (_) {
            var $ = this.convertWhitespaceNodesToSpace(z), J = $.space, Q = $.rawSpace;
            Q !== void 0 && (_.rawSpaceAfter += Q), _.spaces.after += J;
          } else
            z.forEach(function(_e) {
              return v.newNode(_e);
            });
        }
        return;
      }
      var W = this.currToken, ee = void 0;
      L > this.position && (ee = this.parseWhitespaceEquivalentTokens(L));
      var le;
      if (this.isNamedCombinator() ? le = this.namedCombinator() : this.currToken[i.FIELDS.TYPE] === h.combinator ? (le = new w.default({
        value: this.content(),
        source: fe(this.currToken),
        sourceIndex: this.currToken[i.FIELDS.START_POS]
      }), this.position++) : D[this.currToken[i.FIELDS.TYPE]] || ee || this.unexpected(), le) {
        if (ee) {
          var H = this.convertWhitespaceNodesToSpace(ee), ie = H.space, Le = H.rawSpace;
          le.spaces.before = ie, le.rawSpaceBefore = Le;
        }
      } else {
        var Ee = this.convertWhitespaceNodesToSpace(ee, !0), ge = Ee.space, ce = Ee.rawSpace;
        ce || (ce = ge);
        var de = {}, ve = {
          spaces: {}
        };
        ge.endsWith(" ") && ce.endsWith(" ") ? (de.before = ge.slice(0, ge.length - 1), ve.spaces.before = ce.slice(0, ce.length - 1)) : ge.startsWith(" ") && ce.startsWith(" ") ? (de.after = ge.slice(1), ve.spaces.after = ce.slice(1)) : ve.value = ce, le = new w.default({
          value: " ",
          source: j(W, this.tokens[this.position - 1]),
          sourceIndex: W[i.FIELDS.START_POS],
          spaces: de,
          raws: ve
        });
      }
      return this.currToken && this.currToken[i.FIELDS.TYPE] === h.space && (le.spaces.after = this.optionalSpace(this.content()), this.position++), this.newNode(le);
    }, N.comma = function() {
      if (this.position === this.tokens.length - 1) {
        this.root.trailingComma = !0, this.position++;
        return;
      }
      this.current._inferEndPosition();
      var v = new d.default({
        source: {
          start: Z(this.tokens[this.position + 1])
        }
      });
      this.current.parent.append(v), this.current = v, this.position++;
    }, N.comment = function() {
      var v = this.currToken;
      this.newNode(new y.default({
        value: this.content(),
        source: fe(v),
        sourceIndex: v[i.FIELDS.START_POS]
      })), this.position++;
    }, N.error = function(v, L) {
      throw this.root.error(v, L);
    }, N.missingBackslash = function() {
      return this.error("Expected a backslash preceding the semicolon.", {
        index: this.currToken[i.FIELDS.START_POS]
      });
    }, N.missingParenthesis = function() {
      return this.expected("opening parenthesis", this.currToken[i.FIELDS.START_POS]);
    }, N.missingSquareBracket = function() {
      return this.expected("opening square bracket", this.currToken[i.FIELDS.START_POS]);
    }, N.unexpected = function() {
      return this.error("Unexpected '" + this.content() + "'. Escaping special characters with \\ may help.", this.currToken[i.FIELDS.START_POS]);
    }, N.namespace = function() {
      var v = this.prevToken && this.content(this.prevToken) || !0;
      if (this.nextToken[i.FIELDS.TYPE] === h.word)
        return this.position++, this.word(v);
      if (this.nextToken[i.FIELDS.TYPE] === h.asterisk)
        return this.position++, this.universal(v);
    }, N.nesting = function() {
      if (this.nextToken) {
        var v = this.content(this.nextToken);
        if (v === "|") {
          this.position++;
          return;
        }
      }
      var L = this.currToken;
      this.newNode(new A.default({
        value: this.content(),
        source: fe(L),
        sourceIndex: L[i.FIELDS.START_POS]
      })), this.position++;
    }, N.parentheses = function() {
      var v = this.current.last, L = 1;
      if (this.position++, v && v.type === S.PSEUDO) {
        var z = new d.default({
          source: {
            start: Z(this.tokens[this.position - 1])
          }
        }), _ = this.current;
        for (v.append(z), this.current = z; this.position < this.tokens.length && L; )
          this.currToken[i.FIELDS.TYPE] === h.openParenthesis && L++, this.currToken[i.FIELDS.TYPE] === h.closeParenthesis && L--, L ? this.parse() : (this.current.source.end = pe(this.currToken), this.current.parent.source.end = pe(this.currToken), this.position++);
        this.current = _;
      } else {
        for (var $ = this.currToken, J = "(", Q; this.position < this.tokens.length && L; )
          this.currToken[i.FIELDS.TYPE] === h.openParenthesis && L++, this.currToken[i.FIELDS.TYPE] === h.closeParenthesis && L--, Q = this.currToken, J += this.parseParenthesisToken(this.currToken), this.position++;
        v ? v.appendToPropertyAndEscape("value", J, J) : this.newNode(new n.default({
          value: J,
          source: ue($[i.FIELDS.START_LINE], $[i.FIELDS.START_COL], Q[i.FIELDS.END_LINE], Q[i.FIELDS.END_COL]),
          sourceIndex: $[i.FIELDS.START_POS]
        }));
      }
      if (L)
        return this.expected("closing parenthesis", this.currToken[i.FIELDS.START_POS]);
    }, N.pseudo = function() {
      for (var v = this, L = "", z = this.currToken; this.currToken && this.currToken[i.FIELDS.TYPE] === h.colon; )
        L += this.content(), this.position++;
      if (!this.currToken)
        return this.expected(["pseudo-class", "pseudo-element"], this.position - 1);
      if (this.currToken[i.FIELDS.TYPE] === h.word)
        this.splitWord(!1, function(_, $) {
          L += _, v.newNode(new o.default({
            value: L,
            source: j(z, v.currToken),
            sourceIndex: z[i.FIELDS.START_POS]
          })), $ > 1 && v.nextToken && v.nextToken[i.FIELDS.TYPE] === h.openParenthesis && v.error("Misplaced parenthesis.", {
            index: v.nextToken[i.FIELDS.START_POS]
          });
        });
      else
        return this.expected(["pseudo-class", "pseudo-element"], this.currToken[i.FIELDS.START_POS]);
    }, N.space = function() {
      var v = this.content();
      this.position === 0 || this.prevToken[i.FIELDS.TYPE] === h.comma || this.prevToken[i.FIELDS.TYPE] === h.openParenthesis || this.current.nodes.every(function(L) {
        return L.type === "comment";
      }) ? (this.spaces = this.optionalSpace(v), this.position++) : this.position === this.tokens.length - 1 || this.nextToken[i.FIELDS.TYPE] === h.comma || this.nextToken[i.FIELDS.TYPE] === h.closeParenthesis ? (this.current.last.spaces.after = this.optionalSpace(v), this.position++) : this.combinator();
    }, N.string = function() {
      var v = this.currToken;
      this.newNode(new n.default({
        value: this.content(),
        source: fe(v),
        sourceIndex: v[i.FIELDS.START_POS]
      })), this.position++;
    }, N.universal = function(v) {
      var L = this.nextToken;
      if (L && this.content(L) === "|")
        return this.position++, this.namespace();
      var z = this.currToken;
      this.newNode(new f.default({
        value: this.content(),
        source: fe(z),
        sourceIndex: z[i.FIELDS.START_POS]
      }), v), this.position++;
    }, N.splitWord = function(v, L) {
      for (var z = this, _ = this.nextToken, $ = this.content(); _ && ~[h.dollar, h.caret, h.equals, h.word].indexOf(_[i.FIELDS.TYPE]); ) {
        this.position++;
        var J = this.content();
        if ($ += J, J.lastIndexOf("\\") === J.length - 1) {
          var Q = this.nextToken;
          Q && Q[i.FIELDS.TYPE] === h.space && ($ += this.requiredSpace(this.content(Q)), this.position++);
        }
        _ = this.nextToken;
      }
      var W = Se($, ".").filter(function(ie) {
        var Le = $[ie - 1] === "\\", Ee = /^\d+\.\d+%$/.test($);
        return !Le && !Ee;
      }), ee = Se($, "#").filter(function(ie) {
        return $[ie - 1] !== "\\";
      }), le = Se($, "#{");
      le.length && (ee = ee.filter(function(ie) {
        return !~le.indexOf(ie);
      }));
      var H = (0, R.default)(Re([0].concat(W, ee)));
      H.forEach(function(ie, Le) {
        var Ee = H[Le + 1] || $.length, ge = $.slice(ie, Ee);
        if (Le === 0 && L)
          return L.call(z, ge, H.length);
        var ce, de = z.currToken, ve = de[i.FIELDS.START_POS] + H[Le], _e = ue(de[1], de[2] + ie, de[3], de[2] + (Ee - 1));
        if (~W.indexOf(ie)) {
          var we = {
            value: ge.slice(1),
            source: _e,
            sourceIndex: ve
          };
          ce = new g.default(he(we, "value"));
        } else if (~ee.indexOf(ie)) {
          var Fe = {
            value: ge.slice(1),
            source: _e,
            sourceIndex: ve
          };
          ce = new p.default(he(Fe, "value"));
        } else {
          var Be = {
            value: ge,
            source: _e,
            sourceIndex: ve
          };
          he(Be, "value"), ce = new m.default(Be);
        }
        z.newNode(ce, v), v = null;
      }), this.position++;
    }, N.word = function(v) {
      var L = this.nextToken;
      return L && this.content(L) === "|" ? (this.position++, this.namespace()) : this.splitWord(v);
    }, N.loop = function() {
      for (; this.position < this.tokens.length; )
        this.parse(!0);
      return this.current._inferEndPosition(), this.root;
    }, N.parse = function(v) {
      switch (this.currToken[i.FIELDS.TYPE]) {
        case h.space:
          this.space();
          break;
        case h.comment:
          this.comment();
          break;
        case h.openParenthesis:
          this.parentheses();
          break;
        case h.closeParenthesis:
          v && this.missingParenthesis();
          break;
        case h.openSquare:
          this.attribute();
          break;
        case h.dollar:
        case h.caret:
        case h.equals:
        case h.word:
          this.word();
          break;
        case h.colon:
          this.pseudo();
          break;
        case h.comma:
          this.comma();
          break;
        case h.asterisk:
          this.universal();
          break;
        case h.ampersand:
          this.nesting();
          break;
        case h.slash:
        case h.combinator:
          this.combinator();
          break;
        case h.str:
          this.string();
          break;
        case h.closeSquare:
          this.missingSquareBracket();
        case h.semicolon:
          this.missingBackslash();
        default:
          this.unexpected();
      }
    }, N.expected = function(v, L, z) {
      if (Array.isArray(v)) {
        var _ = v.pop();
        v = v.join(", ") + " or " + _;
      }
      var $ = /^[aeiou]/.test(v[0]) ? "an" : "a";
      return z ? this.error("Expected " + $ + " " + v + ', found "' + z + '" instead.', {
        index: L
      }) : this.error("Expected " + $ + " " + v + ".", {
        index: L
      });
    }, N.requiredSpace = function(v) {
      return this.options.lossy ? " " : v;
    }, N.optionalSpace = function(v) {
      return this.options.lossy ? "" : v;
    }, N.lossySpace = function(v, L) {
      return this.options.lossy ? L ? " " : "" : v;
    }, N.parseParenthesisToken = function(v) {
      var L = this.content(v);
      return v[i.FIELDS.TYPE] === h.space ? this.requiredSpace(L) : L;
    }, N.newNode = function(v, L) {
      return L && (/^ +$/.test(L) && (this.options.lossy || (this.spaces = (this.spaces || "") + L), L = !0), v.namespace = L, he(v, "namespace")), this.spaces && (v.spaces.before = this.spaces, this.spaces = ""), this.current.append(v);
    }, N.content = function(v) {
      return v === void 0 && (v = this.currToken), this.css.slice(v[i.FIELDS.START_POS], v[i.FIELDS.END_POS]);
    }, N.locateNextMeaningfulToken = function(v) {
      v === void 0 && (v = this.position + 1);
      for (var L = v; L < this.tokens.length; )
        if (Y[this.tokens[L][i.FIELDS.TYPE]]) {
          L++;
          continue;
        } else
          return L;
      return -1;
    }, I(F, [{
      key: "currToken",
      get: function() {
        return this.tokens[this.position];
      }
    }, {
      key: "nextToken",
      get: function() {
        return this.tokens[this.position + 1];
      }
    }, {
      key: "prevToken",
      get: function() {
        return this.tokens[this.position - 1];
      }
    }]), F;
  }();
  t.default = Oe, a.exports = t.default;
})(_t, _t.exports);
var go = _t.exports;
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = d(go);
  function d(y) {
    return y && y.__esModule ? y : { default: y };
  }
  var g = /* @__PURE__ */ function() {
    function y(m, n) {
      this.func = m || function() {
      }, this.funcRes = null, this.options = n;
    }
    var p = y.prototype;
    return p._shouldUpdateSelector = function(n, o) {
      o === void 0 && (o = {});
      var l = Object.assign({}, this.options, o);
      return l.updateSelector === !1 ? !1 : typeof n != "string";
    }, p._isLossy = function(n) {
      n === void 0 && (n = {});
      var o = Object.assign({}, this.options, n);
      return o.lossless === !1;
    }, p._root = function(n, o) {
      o === void 0 && (o = {});
      var l = new b.default(n, this._parseOptions(o));
      return l.root;
    }, p._parseOptions = function(n) {
      return {
        lossy: this._isLossy(n)
      };
    }, p._run = function(n, o) {
      var l = this;
      return o === void 0 && (o = {}), new Promise(function(f, w) {
        try {
          var A = l._root(n, o);
          Promise.resolve(l.func(A)).then(function(R) {
            var i = void 0;
            return l._shouldUpdateSelector(n, o) && (i = A.toString(), n.selector = i), {
              transform: R,
              root: A,
              string: i
            };
          }).then(f, w);
        } catch (R) {
          w(R);
          return;
        }
      });
    }, p._runSync = function(n, o) {
      o === void 0 && (o = {});
      var l = this._root(n, o), f = this.func(l);
      if (f && typeof f.then == "function")
        throw new Error("Selector processor returned a promise to a synchronous call.");
      var w = void 0;
      return o.updateSelector && typeof n != "string" && (w = l.toString(), n.selector = w), {
        transform: f,
        root: l,
        string: w
      };
    }, p.ast = function(n, o) {
      return this._run(n, o).then(function(l) {
        return l.root;
      });
    }, p.astSync = function(n, o) {
      return this._runSync(n, o).root;
    }, p.transform = function(n, o) {
      return this._run(n, o).then(function(l) {
        return l.transform;
      });
    }, p.transformSync = function(n, o) {
      return this._runSync(n, o).transform;
    }, p.process = function(n, o) {
      return this._run(n, o).then(function(l) {
        return l.string || l.root.toString();
      });
    }, p.processSync = function(n, o) {
      var l = this._runSync(n, o);
      return l.string || l.root.toString();
    }, y;
  }();
  t.default = g, a.exports = t.default;
})(bt, bt.exports);
var vo = bt.exports, Br = {}, re = {};
re.__esModule = !0;
re.universal = re.tag = re.string = re.selector = re.root = re.pseudo = re.nesting = re.id = re.comment = re.combinator = re.className = re.attribute = void 0;
var mo = xe(jt), wo = xe(Tr), bo = xe(Pr), _o = xe(yr), To = xe(Sr), yo = xe(Lr), So = xe(kr), Oo = xe(br), Eo = xe(_r), ko = xe(Er), xo = xe(Or), Po = xe(xr);
function xe(a) {
  return a && a.__esModule ? a : { default: a };
}
var Lo = function(t) {
  return new mo.default(t);
};
re.attribute = Lo;
var Io = function(t) {
  return new wo.default(t);
};
re.className = Io;
var Ao = function(t) {
  return new bo.default(t);
};
re.combinator = Ao;
var Bo = function(t) {
  return new _o.default(t);
};
re.comment = Bo;
var Do = function(t) {
  return new To.default(t);
};
re.id = Do;
var Ro = function(t) {
  return new yo.default(t);
};
re.nesting = Ro;
var Fo = function(t) {
  return new So.default(t);
};
re.pseudo = Fo;
var qo = function(t) {
  return new Oo.default(t);
};
re.root = qo;
var Mo = function(t) {
  return new Eo.default(t);
};
re.selector = Mo;
var No = function(t) {
  return new ko.default(t);
};
re.string = No;
var Co = function(t) {
  return new xo.default(t);
};
re.tag = Co;
var zo = function(t) {
  return new Po.default(t);
};
re.universal = zo;
var X = {};
X.__esModule = !0;
X.isNode = Qt;
X.isPseudoElement = Fr;
X.isPseudoClass = Jo;
X.isContainer = Xo;
X.isNamespace = Zo;
X.isUniversal = X.isTag = X.isString = X.isSelector = X.isRoot = X.isPseudo = X.isNesting = X.isIdentifier = X.isComment = X.isCombinator = X.isClassName = X.isAttribute = void 0;
var se = G, be, Wo = (be = {}, be[se.ATTRIBUTE] = !0, be[se.CLASS] = !0, be[se.COMBINATOR] = !0, be[se.COMMENT] = !0, be[se.ID] = !0, be[se.NESTING] = !0, be[se.PSEUDO] = !0, be[se.ROOT] = !0, be[se.SELECTOR] = !0, be[se.STRING] = !0, be[se.TAG] = !0, be[se.UNIVERSAL] = !0, be);
function Qt(a) {
  return typeof a == "object" && Wo[a.type];
}
function Pe(a, t) {
  return Qt(t) && t.type === a;
}
var Dr = Pe.bind(null, se.ATTRIBUTE);
X.isAttribute = Dr;
var Uo = Pe.bind(null, se.CLASS);
X.isClassName = Uo;
var $o = Pe.bind(null, se.COMBINATOR);
X.isCombinator = $o;
var Ho = Pe.bind(null, se.COMMENT);
X.isComment = Ho;
var Go = Pe.bind(null, se.ID);
X.isIdentifier = Go;
var jo = Pe.bind(null, se.NESTING);
X.isNesting = jo;
var Yt = Pe.bind(null, se.PSEUDO);
X.isPseudo = Yt;
var Qo = Pe.bind(null, se.ROOT);
X.isRoot = Qo;
var Yo = Pe.bind(null, se.SELECTOR);
X.isSelector = Yo;
var Vo = Pe.bind(null, se.STRING);
X.isString = Vo;
var Rr = Pe.bind(null, se.TAG);
X.isTag = Rr;
var Ko = Pe.bind(null, se.UNIVERSAL);
X.isUniversal = Ko;
function Fr(a) {
  return Yt(a) && a.value && (a.value.startsWith("::") || a.value.toLowerCase() === ":before" || a.value.toLowerCase() === ":after" || a.value.toLowerCase() === ":first-letter" || a.value.toLowerCase() === ":first-line");
}
function Jo(a) {
  return Yt(a) && !Fr(a);
}
function Xo(a) {
  return !!(Qt(a) && a.walk);
}
function Zo(a) {
  return Dr(a) || Rr(a);
}
(function(a) {
  a.__esModule = !0;
  var t = G;
  Object.keys(t).forEach(function(g) {
    g === "default" || g === "__esModule" || g in a && a[g] === t[g] || (a[g] = t[g]);
  });
  var b = re;
  Object.keys(b).forEach(function(g) {
    g === "default" || g === "__esModule" || g in a && a[g] === b[g] || (a[g] = b[g]);
  });
  var d = X;
  Object.keys(d).forEach(function(g) {
    g === "default" || g === "__esModule" || g in a && a[g] === d[g] || (a[g] = d[g]);
  });
})(Br);
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = p(vo), d = y(Br);
  function g() {
    if (typeof WeakMap != "function")
      return null;
    var o = /* @__PURE__ */ new WeakMap();
    return g = function() {
      return o;
    }, o;
  }
  function y(o) {
    if (o && o.__esModule)
      return o;
    if (o === null || typeof o != "object" && typeof o != "function")
      return { default: o };
    var l = g();
    if (l && l.has(o))
      return l.get(o);
    var f = {}, w = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var A in o)
      if (Object.prototype.hasOwnProperty.call(o, A)) {
        var R = w ? Object.getOwnPropertyDescriptor(o, A) : null;
        R && (R.get || R.set) ? Object.defineProperty(f, A, R) : f[A] = o[A];
      }
    return f.default = o, l && l.set(o, f), f;
  }
  function p(o) {
    return o && o.__esModule ? o : { default: o };
  }
  var m = function(l) {
    return new b.default(l);
  };
  Object.assign(m, d), delete m.__esModule;
  var n = m;
  t.default = n, a.exports = t.default;
})(wt, wt.exports);
var ea = wt.exports;
const ta = li, qr = ea, ra = qr();
var na = {
  isUsableColor(a, t) {
    return ta(t) && a !== "gray" && t[600];
  },
  /**
   * @param {string} selector
   */
  commonTrailingPseudos(a) {
    let t = ra.astSync(a), b = [];
    for (let [g, y] of t.nodes.entries())
      for (const [p, m] of [...y.nodes].reverse().entries()) {
        if (m.type !== "pseudo" || !m.value.startsWith("::"))
          break;
        b[p] = b[p] || [], b[p][g] = m;
      }
    let d = qr.selector();
    for (const g of b) {
      if (!g)
        continue;
      if ((/* @__PURE__ */ new Set([...g.map((p) => p.value)])).size > 1)
        break;
      g.forEach((p) => p.remove()), d.prepend(g[0]);
    }
    return d.nodes.length ? [d.toString(), t.toString()] : [null, a];
  }
};
const ia = Gn, oa = jn, aa = Vn, sa = Xn, { commonTrailingPseudos: ua } = na, lr = {
  // Reserved for future "magic properties", for example:
  // bulletColor: (color) => ({ 'ul > li::before': { backgroundColor: color } }),
};
function zt(a, { className: t, modifier: b, prefix: d }) {
  let g = d(`.not-${t}`).slice(1), y = a.startsWith(">") ? `${b === "DEFAULT" ? `.${t}` : `.${t}-${b}`} ` : "", [p, m] = ua(a);
  return p ? `:where(${y}${m}):not(:where([class~="${g}"],[class~="${g}"] *))${p}` : `:where(${y}${a}):not(:where([class~="${g}"],[class~="${g}"] *))`;
}
function fr(a) {
  return typeof a == "object" && a !== null;
}
function la(a = {}, { target: t, className: b, modifier: d, prefix: g }) {
  function y(p, m) {
    return t === "legacy" ? [p, m] : Array.isArray(m) ? [p, m] : fr(m) ? Object.values(m).some(fr) ? [
      zt(p, { className: b, modifier: d, prefix: g }),
      m,
      Object.fromEntries(Object.entries(m).map(([o, l]) => y(o, l)))
    ] : [zt(p, { className: b, modifier: d, prefix: g }), m] : [p, m];
  }
  return Object.fromEntries(
    Object.entries(
      oa(
        {},
        ...Object.keys(a).filter((p) => lr[p]).map((p) => lr[p](a[p])),
        ...aa(a.css || {})
      )
    ).map(([p, m]) => y(p, m))
  );
}
var fa = ia.withOptions(
  ({ className: a = "prose", target: t = "modern" } = {}) => function({ addVariant: b, addComponents: d, theme: g, prefix: y }) {
    let p = g("typography"), m = { className: a, prefix: y };
    for (let [n, ...o] of [
      ["headings", "h1", "h2", "h3", "h4", "h5", "h6", "th"],
      ["h1"],
      ["h2"],
      ["h3"],
      ["h4"],
      ["h5"],
      ["h6"],
      ["p"],
      ["a"],
      ["blockquote"],
      ["figure"],
      ["figcaption"],
      ["strong"],
      ["em"],
      ["code"],
      ["pre"],
      ["ol"],
      ["ul"],
      ["li"],
      ["table"],
      ["thead"],
      ["tr"],
      ["th"],
      ["td"],
      ["img"],
      ["video"],
      ["hr"],
      ["lead", '[class~="lead"]']
    ]) {
      o = o.length === 0 ? [n] : o;
      let l = t === "legacy" ? o.map((f) => `& ${f}`) : o.join(", ");
      b(
        `${a}-${n}`,
        t === "legacy" ? l : `& :is(${zt(l, m)})`
      );
    }
    d(
      Object.keys(p).map((n) => ({
        [n === "DEFAULT" ? `.${a}` : `.${a}-${n}`]: la(
          p[n],
          {
            target: t,
            className: a,
            modifier: n,
            prefix: y
          }
        )
      }))
    );
  },
  () => ({
    theme: { typography: sa }
  })
);
const ca = /* @__PURE__ */ Hn(fa), ha = {
  content: [],
  future: {
    hoverOnlyWhenSupported: !0
  },
  theme: {
    extend: {
      typography: ({ theme: a }) => ({
        DEFAULT: {
          css: [
            {
              color: a("colors.neutral.900"),
              fontSize: "16px",
              fontFamily: a("fontFamily.body"),
              p: {
                marginTop: "20px",
                marginBottom: "20px"
              },
              '[class~="lead"]': {
                fontWeight: a("fontWeight.medium"),
                fontSize: "20px",
                lineHeight: "32px",
                marginTop: "24px",
                marginBottom: "24px",
                color: a("colors.neutral.900")
              },
              blockquote: {
                fontWeight: a("fontWeight.medium"),
                color: a("colors.neutral.900"),
                marginTop: "32px",
                marginBottom: "32px",
                paddingLeft: "20px"
              },
              "blockquote ~ figcaption ": {
                fontWeight: a("fontWeight.medium"),
                color: a("colors.neutral.900"),
                fontStyle: "italic"
              },
              h1: {
                fontSize: "36px",
                fontFamily: a("fontFamily.headings"),
                marginTop: "0",
                marginBottom: "32px",
                lineHeight: "40px"
              },
              h2: {
                fontSize: "24px",
                fontFamily: a("fontFamily.headings"),
                marginTop: "48px",
                marginBottom: "24px",
                lineHeight: "32px"
              },
              h3: {
                fontSize: "20px",
                fontFamily: a("fontFamily.headings"),
                marginTop: "32px",
                marginBottom: "12px",
                lineHeight: "32px"
              },
              h4: {
                fontFamily: a("fontFamily.headings"),
                marginTop: "24px",
                marginBottom: "8px",
                lineHeight: "24px"
              },
              img: {
                marginTop: "32px",
                marginBottom: "32px"
              },
              video: {
                marginTop: "32px",
                marginBottom: "32px"
              },
              figure: {
                marginTop: "32px",
                marginBottom: "32px"
              },
              "figure > *": {
                marginTop: "0",
                marginBottom: "0"
              },
              figcaption: {
                fontWeight: a("fontStyle.italic"),
                fontSize: "14px",
                lineHeight: "20px",
                marginTop: "12px"
              },
              code: {
                fontSize: "14px"
              },
              "h2 code": {
                fontSize: "21px"
              },
              "h3 code": {
                fontSize: "18px"
              },
              strong: {
                fontWeight: a("fontWeight.medium")
              },
              pre: {
                fontSize: "14px",
                lineHeight: "24px",
                marginTop: "24px",
                marginBottom: "24px",
                borderRadius: "6px",
                paddingTop: "12px",
                paddingRight: "16px",
                paddingBottom: "12px",
                paddingLeft: "16px"
              },
              ol: {
                marginTop: "20px",
                marginBottom: "20px",
                paddingLeft: "20px"
              },
              ul: {
                marginTop: "20px",
                marginBottom: "20px",
                paddingLeft: "26px"
              },
              li: {
                marginTop: "8px",
                marginBottom: "8px"
              },
              "ol > li": {
                paddingLeft: "6px"
              },
              "ul > li": {
                paddingLeft: "6px"
              },
              "> ul > li p": {
                marginTop: "12px",
                marginBottom: "12px"
              },
              "> ul > li > *:first-child": {
                marginTop: "20px"
              },
              "> ul > li > *:last-child": {
                marginBottom: "20px"
              },
              "> ol > li > *:first-child": {
                marginTop: "20px"
              },
              "> ol > li > *:last-child": {
                marginBottom: "20px"
              },
              "ul ul, ul ol, ol ul, ol ol": {
                marginTop: "12px",
                marginBottom: "12px"
              },
              hr: {
                marginTop: "48px",
                marginBottom: "48px"
              },
              "hr + *": {
                marginTop: "0"
              },
              "h2 + *": {
                marginTop: "0"
              },
              "h3 + *": {
                marginTop: "0"
              },
              "h4 + *": {
                marginTop: "0"
              },
              table: {
                fontSize: "14px",
                lineHeight: "24px"
              },
              "thead th": {
                fontWeight: a("fontWeight.medium"),
                paddingRight: "8px",
                paddingBottom: "8px",
                paddingLeft: "8px"
              },
              "thead th:first-child": {
                paddingLeft: "0"
              },
              "thead th:last-child": {
                paddingRight: "0"
              },
              "tbody td, tfoot td": {
                paddingTop: "8px",
                paddingRight: "8px",
                paddingBottom: "8px",
                paddingLeft: "8px"
              },
              "tbody td:first-child, tfoot td:first-child": {
                paddingLeft: "0"
              },
              "tbody td:last-child, tfoot td:last-child": {
                paddingRight: "0"
              }
            },
            {
              "> :first-child": {
                marginTop: "0"
              },
              "> :last-child": {
                marginBottom: "0"
              }
            }
          ]
        }
      }),
      outlineColor: ({ theme: a }) => ({
        DEFAULT: a("colors.secondary.600")
      }),
      outlineOffset: {
        DEFAULT: "2px"
      },
      outlineWidth: {
        DEFAULT: "2px"
      },
      boxShadow: {
        DEFAULT: "0px 1px 3px rgba(0, 0, 0, 0.1), 0px 1px 2px rgba(0, 0, 0, 0.06)",
        md: "0px 4px 6px -1px rgba(0, 0, 0, 0.1), 0px 2px 4px -1px rgba(0, 0, 0, 0.06)"
      },
      backgroundImage: {
        "checked-checkbox-current": "linear-gradient(-45deg,transparent 65%, currentcolor 65.99%),linear-gradient(45deg,transparent 75%, currentcolor 75.99%),linear-gradient(-45deg, currentcolor 40%,transparent 40.99%),linear-gradient(45deg, currentcolor 30%, white 30.99%, white 40%,transparent 40.99%),linear-gradient(-45deg, white 50%, currentcolor 50.99%)",
        "indeterminate-checkbox-current": "linear-gradient(90deg,transparent 80%, currentcolor 80%),linear-gradient(-90deg,transparent 80%, currentcolor 80%),linear-gradient(0deg, currentcolor 43%, white 43%, white 57%, currentcolor 57%)"
      },
      colors: {
        brand: "rgb(var(--colors-brand) / <alpha-value>)",
        neutral: {
          900: "rgb(var(--colors-neutral-900) / <alpha-value>)",
          800: "rgb(var(--colors-neutral-800) / <alpha-value>)",
          700: "rgb(var(--colors-neutral-700) / <alpha-value>)",
          600: "rgb(var(--colors-neutral-600) / <alpha-value>)",
          500: "rgb(var(--colors-neutral-500) / <alpha-value>)",
          400: "rgb(var(--colors-neutral-400) / <alpha-value>)",
          300: "rgb(var(--colors-neutral-300) / <alpha-value>)",
          200: "rgb(var(--colors-neutral-200) / <alpha-value>)",
          100: "rgb(var(--colors-neutral-100) / <alpha-value>)",
          50: "rgb(var(--colors-neutral-50) / <alpha-value>)"
        },
        primary: {
          900: "rgb(var(--colors-primary-900) / <alpha-value>)",
          800: "rgb(var(--colors-primary-800) / <alpha-value>)",
          700: "rgb(var(--colors-primary-700) / <alpha-value>)",
          600: "rgb(var(--colors-primary-600) / <alpha-value>)",
          500: "rgb(var(--colors-primary-500) / <alpha-value>)",
          400: "rgb(var(--colors-primary-400) / <alpha-value>)",
          300: "rgb(var(--colors-primary-300) / <alpha-value>)",
          200: "rgb(var(--colors-primary-200) / <alpha-value>)",
          100: "rgb(var(--colors-primary-100) / <alpha-value>)",
          50: "rgb(var(--colors-primary-50) / <alpha-value>)"
        },
        secondary: {
          900: "rgb(var(--colors-secondary-900) / <alpha-value>)",
          800: "rgb(var(--colors-secondary-800) / <alpha-value>)",
          700: "rgb(var(--colors-secondary-700) / <alpha-value>)",
          600: "rgb(var(--colors-secondary-600) / <alpha-value>)",
          500: "rgb(var(--colors-secondary-500) / <alpha-value>)",
          400: "rgb(var(--colors-secondary-400) / <alpha-value>)",
          300: "rgb(var(--colors-secondary-300) / <alpha-value>)",
          200: "rgb(var(--colors-secondary-200) / <alpha-value>)",
          100: "rgb(var(--colors-secondary-100) / <alpha-value>)",
          50: "rgb(var(--colors-secondary-50) / <alpha-value>)"
        },
        positive: {
          900: "rgb(var(--colors-positive-900) / <alpha-value>)",
          800: "rgb(var(--colors-positive-800) / <alpha-value>)",
          700: "rgb(var(--colors-positive-700) / <alpha-value>)",
          600: "rgb(var(--colors-positive-600) / <alpha-value>)",
          500: "rgb(var(--colors-positive-500) / <alpha-value>)",
          400: "rgb(var(--colors-positive-400) / <alpha-value>)",
          300: "rgb(var(--colors-positive-300) / <alpha-value>)",
          200: "rgb(var(--colors-positive-200) / <alpha-value>)",
          100: "rgb(var(--colors-positive-100) / <alpha-value>)",
          50: "rgb(var(--colors-positive-50) / <alpha-value>)"
        },
        negative: {
          900: "rgb(var(--colors-negative-900) / <alpha-value>)",
          800: "rgb(var(--colors-negative-800) / <alpha-value>)",
          700: "rgb(var(--colors-negative-700) / <alpha-value>)",
          600: "rgb(var(--colors-negative-600) / <alpha-value>)",
          500: "rgb(var(--colors-negative-500) / <alpha-value>)",
          400: "rgb(var(--colors-negative-400) / <alpha-value>)",
          300: "rgb(var(--colors-negative-300) / <alpha-value>)",
          200: "rgb(var(--colors-negative-200) / <alpha-value>)",
          100: "rgb(var(--colors-negative-100) / <alpha-value>)",
          50: "rgb(var(--colors-negative-50) / <alpha-value>)"
        },
        warning: {
          900: "rgb(var(--colors-warning-900) / <alpha-value>)",
          800: "rgb(var(--colors-warning-800) / <alpha-value>)",
          700: "rgb(var(--colors-warning-700) / <alpha-value>)",
          600: "rgb(var(--colors-warning-600) / <alpha-value>)",
          500: "rgb(var(--colors-warning-500) / <alpha-value>)",
          400: "rgb(var(--colors-warning-400) / <alpha-value>)",
          300: "rgb(var(--colors-warning-300) / <alpha-value>)",
          200: "rgb(var(--colors-warning-200) / <alpha-value>)",
          100: "rgb(var(--colors-warning-100) / <alpha-value>)",
          50: "rgb(var(--colors-warning-50) / <alpha-value>)"
        },
        disabled: {
          900: "rgb(var(--colors-disabled-900) / 0.5)",
          800: "rgb(var(--colors-disabled-800) / 0.5)",
          700: "rgb(var(--colors-disabled-700) / 0.5)",
          600: "rgb(var(--colors-disabled-600) / 0.5)",
          500: "rgb(var(--colors-disabled-500) / 0.5)",
          400: "rgb(var(--colors-disabled-400) / 0.5)",
          300: "rgb(var(--colors-disabled-300) / 0.5)",
          200: "rgb(var(--colors-disabled-200) / 0.5)",
          100: "rgb(var(--colors-disabled-100) / 0.5)",
          50: "rgb(var(--colors-disabled-50) / 0.5)"
        }
      },
      fontFamily: {
        headings: "Inter, sans-serif",
        body: "Inter, sans-serif"
      },
      screens: {
        xs: "376px",
        sm: "640px"
      },
      keyframes: {
        "stroke-loader-circular": {
          "0%": { "stroke-dasharray": "1, 200", "stroke-dashoffset": "0" },
          "50%": { "stroke-dasharray": "140, 200", "stroke-dashoffset": "-35" },
          "100%": { "stroke-dasharray": "89, 200", "stroke-dashoffset": "-150" }
        },
        line: {
          from: {
            left: "-100%",
            width: "100%"
          },
          to: {
            left: "100%",
            width: "10%"
          }
        }
      },
      animation: {
        "spin-slow": "spin 1.5s linear infinite",
        line: "line 1.5s ease-in infinite",
        "stroke-loader-circular": "stroke-loader-circular 2s ease-in-out infinite"
      },
      fontSize: {
        "2xs": ["10px", "11px"],
        "3xs": ["8px", "8px"]
      }
    },
    variables: {
      DEFAULT: {
        colors: {
          primary: {
            50: "235 255 242",
            100: "217 253 228",
            200: "171 241 192",
            300: "130 234 158",
            400: "60 224 120",
            500: "19 195 96",
            600: "7 161 79",
            700: "13 127 63",
            800: "16 86 46",
            900: "15 50 30"
          },
          secondary: {
            50: "247 245 253",
            100: "239 236 251",
            200: "225 218 246",
            300: "216 203 245",
            400: "198 177 246",
            500: "180 151 249",
            600: "151 111 238",
            700: "119 79 209",
            800: "82 52 153",
            900: "48 32 86"
          },
          positive: {
            50: "235 255 242",
            100: "217 253 228",
            200: "171 241 192",
            300: "130 234 158",
            400: "60 224 120",
            500: "19 195 96",
            600: "7 161 79",
            700: "13 127 63",
            800: "16 86 46",
            900: "15 50 30"
          },
          negative: {
            50: "255 245 247",
            100: "255 232 237",
            200: "254 211 219",
            300: "253 193 202",
            400: "255 163 175",
            500: "255 127 143",
            600: "240 59 91",
            700: "208 13 48",
            800: "141 8 33",
            900: "76 15 23"
          },
          warning: {
            50: "254 247 236",
            100: "255 238 211",
            200: "254 220 165",
            300: "254 202 132",
            400: "255 181 77",
            500: "237 153 14",
            600: "191 121 17",
            700: "157 93 3",
            800: "109 63 9",
            900: "62 35 10"
          },
          neutral: {
            50: "249 251 250",
            100: "239 244 241",
            200: "217 226 220",
            300: "187 198 190",
            400: "129 140 133",
            500: "100 111 104",
            600: "77 86 79",
            700: "56 65 59",
            800: "37 43 39",
            900: "21 26 22"
          },
          disabled: {
            50: "249 251 250",
            100: "239 244 241",
            200: "217 226 220",
            300: "187 198 190",
            400: "129 140 133",
            500: "100 111 104",
            600: "77 86 79",
            700: "56 65 59",
            800: "37 43 39",
            900: "21 26 22"
          }
        }
      }
    }
  },
  plugins: [ca, Un, $n]
};
export {
  ha as tailwindConfig
};
