// @storefront-ui/vue/dist/components/SfScrollable/SfScrollable.vue.mjs
import { defineComponent as $, toRefs as w, computed as u, reactive as A, openBlock as c, createElementBlock as E, normalizeClass as I, unref as t, renderSlot as v, normalizeProps as p, mergeProps as s, createBlock as m, withCtx as g, createVNode as x, createCommentVNode as y, resolveDynamicComponent as L } from "vue";
import { SfScrollableDirection as P, SfScrollableButtonsPlacement as o } from "@storefront-ui/shared";
import { useScrollable as R } from "../../composables/useScrollable/useScrollable.mjs";
import S from "../SfButton/SfButton.vue.mjs";
import j from "../SfIcons/SfIconChevronLeft.vue.mjs";
import q from "../SfIcons/SfIconChevronRight.vue.mjs";
import { ClassProp as M } from "../../shared/props.mjs";
const O = {
  inheritAttrs: !1
}, U = /* @__PURE__ */ $({
  ...O,
  __name: "SfScrollable",
  props: {
    tag: {
      type: [String, Object],
      default: "div"
    },
    direction: {
      type: String,
      default: P.horizontal
    },
    buttonsPlacement: {
      type: String,
      default: o.block
    },
    wrapperClass: M,
    activeIndex: {
      type: Number,
      default: void 0
    },
    reduceMotion: {
      type: Boolean,
      default: void 0
    },
    drag: {
      type: [Object, Boolean],
      default: void 0
    },
    prevDisabled: {
      type: Boolean,
      default: void 0
    },
    nextDisabled: {
      type: Boolean,
      default: void 0
    },
    isActiveIndexCentered: {
      type: Boolean,
      default: !1
    },
    buttonPrevAriaLabel: {
      type: String,
      default: "Previous"
    },
    buttonNextAriaLabel: {
      type: String,
      default: "Next"
    }
  },
  emits: ["onDragStart", "onDragEnd", "onScroll", "onPrev", "onNext"],
  setup(a, { emit: r }) {
    const i = a, { direction: B, activeIndex: h, reduceMotion: D, drag: k, isActiveIndexCentered: z } = w(i), { containerRef: C, state: N, getNextButtonProps: b, getPrevButtonProps: f } = R(
      u(() => ({
        ...A({
          direction: B,
          activeIndex: h,
          reduceMotion: D,
          drag: k,
          isActiveIndexCentered: z
        }),
        onDragStart: (e) => r("onDragStart", e),
        onDragEnd: (e) => r("onDragEnd", e),
        onScroll: (e) => r("onScroll", e),
        onPrev: (e) => r("onPrev", e),
        onNext: (e) => r("onNext", e)
      }))
    ), l = u(() => i.direction === P.horizontal), n = u(() => i.buttonsPlacement === o.floating), d = u(() => i.buttonsPlacement === o.block);
    return (e, V) => (c(), E("div", {
      class: I(["items-center", "relative", l.value ? "flex" : "flex-col h-full inline-flex", a.wrapperClass])
    }, [
      e.$slots.previousButton && a.buttonsPlacement !== t(o).none ? v(e.$slots, "previousButton", p(s({ key: 0 }, t(f)))) : a.buttonsPlacement !== t(o).none ? (c(), m(t(S), s({
        key: 1,
        variant: "secondary",
        size: "lg",
        square: "",
        class: [
          "!rounded-full bg-white hidden md:block !ring-neutral-500 !text-neutral-500",
          {
            "mr-4": d.value && l.value,
            "mb-4 rotate-90": d.value && !l.value,
            "absolute left-4 z-10": n.value && l.value,
            "absolute top-4 rotate-90 z-10": n.value && !l.value
          },
          n.value ? "disabled:hidden" : "disabled:!ring-disabled-300 disabled:!text-disabled-500"
        ]
      }, t(f), {
        disabled: a.prevDisabled || t(f).disabled,
        "aria-label": a.buttonPrevAriaLabel
      }), {
        default: g(() => [
          x(t(j))
        ]),
        _: 1
      }, 16, ["class", "disabled", "aria-label"])) : y("", !0),
      (c(), m(L(a.tag), s({
        ref_key: "containerRef",
        ref: C,
        class: [
          "motion-safe:scroll-smooth",
          {
            "overflow-x-auto flex gap-4": l.value,
            "overflow-y-auto flex flex-col gap-4": !l.value,
            "cursor-grab": t(N).isDragged
          }
        ]
      }, { ...e.$attrs, ...i }, { disabled: a.prevDisabled }), {
        default: g(() => [
          v(e.$slots, "default")
        ]),
        _: 3
      }, 16, ["class", "disabled"])),
      e.$slots.nextButton && a.buttonsPlacement !== t(o).none ? v(e.$slots, "nextButton", p(s({ key: 2 }, t(b)))) : a.buttonsPlacement !== t(o).none ? (c(), m(t(S), s({
        key: 3,
        variant: "secondary",
        size: "lg",
        square: "",
        class: [
          "!rounded-full bg-white hidden md:block !ring-neutral-500 !text-neutral-500",
          {
            "ml-4": d.value && l.value,
            "mt-4 rotate-90": d.value && !l.value,
            "absolute right-4 z-10": n.value && l.value,
            "absolute bottom-4 rotate-90 z-10": n.value && !l.value
          },
          n.value ? "disabled:hidden" : "disabled:!ring-disabled-300 disabled:!text-disabled-500"
        ]
      }, t(b), {
        disabled: a.nextDisabled || t(b).disabled,
        "aria-label": a.buttonNextAriaLabel
      }), {
        default: g(() => [
          x(t(q))
        ]),
        _: 1
      }, 16, ["class", "disabled", "aria-label"])) : y("", !0)
    ], 2));
  }
});
export {
  U as default
};
