// @storefront-ui/vue/dist/components/SfSelect/SfSelect.vue.mjs
import { defineComponent as y, computed as v, openBlock as u, createElementBlock as m, normalizeClass as d, unref as e, withDirectives as S, createElementVNode as x, mergeProps as V, withKeys as w, toDisplayString as z, createCommentVNode as C, renderSlot as f, vModelSelect as h, createVNode as B } from "vue";
import N from "../SfIcons/SfIconExpandMore.vue.mjs";
import { SfSelectSize as s } from "@storefront-ui/shared";
import { useDisclosure as k } from "../../composables/useDisclosure/useDisclosure.mjs";
import { useFocusVisible as q } from "../../composables/useFocusVisible/useFocusVisible.mjs";
const $ = ["required", "disabled"], D = {
  inheritAttrs: !1
}, M = /* @__PURE__ */ y({
  ...D,
  __name: "SfSelect",
  props: {
    size: {
      type: String,
      default: s.base
    },
    placeholder: {
      type: String,
      default: ""
    },
    required: {
      type: Boolean,
      default: !1
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    invalid: {
      type: Boolean,
      default: !1
    },
    modelValue: {
      type: String,
      default: ""
    },
    wrapperClassName: {
      type: String,
      default: ""
    }
  },
  emits: ["update:modelValue"],
  setup(t, { emit: c }) {
    const p = t, { isOpen: g, close: n, open: r } = k(), { isFocusVisible: b } = q(), o = v({
      get: () => p.modelValue,
      set: (i) => c("update:modelValue", i)
    });
    return (i, l) => (u(), m("span", {
      class: d([
        "relative flex flex-col rounded-md",
        {
          "focus-within:outline focus-within:outline-offset": e(b)
        },
        t.wrapperClassName
      ]),
      "data-testid": "select"
    }, [
      S(x("select", V({
        required: t.required,
        "onUpdate:modelValue": l[0] || (l[0] = (a) => o.value = a),
        disabled: t.disabled,
        class: [
          "appearance-none disabled:cursor-not-allowed cursor-pointer pl-4 pr-3.5 text-neutral-900 ring-inset focus:ring-primary-700 focus:ring-2 outline-none bg-transparent rounded-md ring-1 ring-neutral-300 hover:ring-primary-700 active:ring-2 active:ring-primary-700 disabled:bg-disabled-100 disabled:text-disabled-900 disabled:ring-disabled-200",
          {
            "py-1.5": t.size === e(s).sm,
            "py-2": t.size === e(s).base,
            "py-3 text-base": t.size === e(s).lg,
            "!ring-negative-700 ring-2": t.invalid && !t.disabled
          }
        ],
        "data-testid": "select-input",
        onBlur: l[1] || (l[1] = //@ts-ignore
        (...a) => e(n) && e(n)(...a)),
        onChange: l[2] || (l[2] = //@ts-ignore
        (...a) => e(n) && e(n)(...a)),
        onClick: l[3] || (l[3] = //@ts-ignore
        (...a) => e(r) && e(r)(...a)),
        onKeydown: l[4] || (l[4] = w(
          //@ts-ignore
          (...a) => e(r) && e(r)(...a),
          ["space"]
        ))
      }, i.$attrs), [
        t.placeholder ? (u(), m("option", {
          key: 0,
          disabled: "",
          hidden: "",
          class: d(["text-sm bg-neutral-300", [
            "bg-neutral-300 text-sm",
            {
              "text-base": t.size === e(s).lg
            }
          ]]),
          value: "",
          "data-testid": "select-placeholder"
        }, z(t.placeholder), 3)) : C("", !0),
        f(i.$slots, "default")
      ], 16, $), [
        [h, o.value]
      ]),
      f(i.$slots, "chevron", {}, () => [
        B(e(N), {
          class: d([
            "absolute -translate-y-1 pointer-events-none top-1/3 right-4 transition easy-in-out duration-0.5",
            t.disabled ? "text-disabled-500" : "text-neutral-500",
            e(g) ? "rotate-180" : ""
          ])
        }, null, 8, ["class"])
      ])
    ], 2));
  }
});
export {
  M as default
};
