// @storefront-ui/vue/dist/components/SfCheckbox/SfCheckbox.vue.mjs
import { defineComponent as l, toRefs as c, computed as b, withDirectives as m, openBlock as p, createElementBlock as s, normalizeClass as u, vModelCheckbox as x } from "vue";
const h = /* @__PURE__ */ l({
  __name: "SfCheckbox",
  props: {
    modelValue: {
      type: [String, Array, Boolean],
      default: !1
    },
    invalid: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["update:modelValue"],
  setup(t, { emit: o }) {
    const d = t, { modelValue: e } = c(d), r = b({
      get: () => e == null ? void 0 : e.value,
      set: (a) => o("update:modelValue", a)
    });
    return (a, n) => m((p(), s("input", {
      "onUpdate:modelValue": n[0] || (n[0] = (i) => r.value = i),
      class: u(["h-[18px] min-w-[18px] border-2 rounded-sm appearance-none cursor-pointer text-gray-500 hover:indeterminate:text-primary-800 enabled:active:checked:text-primary-900 checked:text-primary-700 checked:bg-checked-checkbox-current border-current indeterminate:bg-indeterminate-checkbox-current indeterminate:text-primary-700 disabled:text-gray-300 hover:text-gray-300 disabled:cursor-not-allowed enabled:hover:border-primary-800 enabled:active:border-primary-900 enabled:hover:checked:text-primary-800 enabled:hover:indeterminate:text-primary-800 enabled:checked:text-primary-700 enabled:indeterminate:text-primary-700 enabled:focus-visible:outline enabled:focus-visible:outline-offset", {
        "border-negative-700 enabled:hover:border-negative-800 enabled:active:border-negative-900 indeterminate:bg-none": t.invalid
      }]),
      type: "checkbox",
      "data-testid": "checkbox"
    }, null, 2)), [
      [x, r.value]
    ]);
  }
});
export {
  h as default
};
