// @storefront-ui/vue/dist/components/SfInput/SfInput.vue.mjs
import { defineComponent as m, toRefs as g, ref as w, computed as v, openBlock as b, createBlock as y, resolveDynamicComponent as h, normalizeClass as x, unref as i, withCtx as S, renderSlot as o, withDirectives as V, createElementVNode as C, mergeProps as z, vModelDynamic as B } from "vue";
import { useFocusVisible as D } from "../../composables/useFocusVisible/useFocusVisible.mjs";
import { SfInputSize as a } from "@storefront-ui/shared";
const I = {
  inheritAttrs: !1
}, T = {
  [a.sm]: " h-[32px]",
  [a.base]: "h-[40px]",
  [a.lg]: "h-[48px]"
}, F = /* @__PURE__ */ m({
  ...I,
  __name: "SfInput",
  props: {
    modelValue: {
      type: [String, Number],
      default: ""
    },
    wrapperTag: {
      type: [String, Object],
      default: "span"
    },
    size: {
      type: String,
      default: a.base
    },
    invalid: {
      type: Boolean,
      default: !1
    },
    wrapperClass: {
      type: [String, Object],
      default: ""
    }
  },
  emits: ["update:modelValue"],
  setup(t, { emit: u }) {
    const p = t, { modelValue: d, invalid: n } = g(p), { isFocusVisible: c } = D({ isTextInput: !0 }), r = w(), s = v({
      get: () => d.value ?? r.value,
      set: (e) => {
        u("update:modelValue", e), r.value = e;
      }
    });
    return (e, l) => (b(), y(h(t.wrapperTag), {
      class: x([
        "flex items-center gap-2 px-4 bg-white rounded-md ring-1 text-neutral-500 hover:ring-primary-700 focus-within:caret-primary-700 active:caret-primary-700 active:ring-primary-700 active:ring-2 focus-within:ring-primary-700 focus-within:ring-2",
        {
          "ring-2 ring-negative-700": i(n),
          "ring-1 ring-neutral-200": !i(n),
          "focus-within:outline focus-within:outline-offset": i(c)
        },
        T[t.size],
        t.wrapperClass
      ]),
      "data-testid": "input"
    }, {
      default: S(() => [
        o(e.$slots, "prefix"),
        V(C("input", z({
          "onUpdate:modelValue": l[0] || (l[0] = (f) => s.value = f),
          class: "min-w-[80px] w-full text-base outline-none appearance-none text-neutral-900 disabled:cursor-not-allowed disabled:bg-transparent read-only:bg-transparent",
          size: 1,
          "data-testid": "input-field"
        }, e.$attrs), null, 16), [
          [B, s.value]
        ]),
        o(e.$slots, "suffix")
      ]),
      _: 3
    }, 8, ["class"]));
  }
});
export {
  F as default
};
